{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.66\n"
     ]
    }
   ],
   "source": [
    "import Bio\n",
    "# http://biopython.org/wiki/Main_Page\n",
    "print(Bio.__version__)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "print(help(Bio))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Sequences #"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on class Seq in module Bio.Seq:\n",
      "\n",
      "class Seq(builtins.object)\n",
      " |  A read-only sequence object (essentially a string with an alphabet).\n",
      " |  \n",
      " |  Like normal python strings, our basic sequence object is immutable.\n",
      " |  This prevents you from doing my_seq[5] = \"A\" for example, but does allow\n",
      " |  Seq objects to be used as dictionary keys.\n",
      " |  \n",
      " |  The Seq object provides a number of string like methods (such as count,\n",
      " |  find, split and strip), which are alphabet aware where appropriate.\n",
      " |  \n",
      " |  In addition to the string like sequence, the Seq object has an alphabet\n",
      " |  property. This is an instance of an Alphabet class from Bio.Alphabet,\n",
      " |  for example generic DNA, or IUPAC DNA. This describes the type of molecule\n",
      " |  (e.g. RNA, DNA, protein) and may also indicate the expected symbols\n",
      " |  (letters).\n",
      " |  \n",
      " |  The Seq object also provides some biological methods, such as complement,\n",
      " |  reverse_complement, transcribe, back_transcribe and translate (which are\n",
      " |  not applicable to sequences with a protein alphabet).\n",
      " |  \n",
      " |  Methods defined here:\n",
      " |  \n",
      " |  __add__(self, other)\n",
      " |      Add another sequence or string to this sequence.\n",
      " |      \n",
      " |      If adding a string to a Seq, the alphabet is preserved:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_protein\n",
      " |      >>> Seq(\"MELKI\", generic_protein) + \"LV\"\n",
      " |      Seq('MELKILV', ProteinAlphabet())\n",
      " |      \n",
      " |      When adding two Seq (like) objects, the alphabets are important.\n",
      " |      Consider this example:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet.IUPAC import unambiguous_dna, ambiguous_dna\n",
      " |      >>> unamb_dna_seq = Seq(\"ACGT\", unambiguous_dna)\n",
      " |      >>> ambig_dna_seq = Seq(\"ACRGT\", ambiguous_dna)\n",
      " |      >>> unamb_dna_seq\n",
      " |      Seq('ACGT', IUPACUnambiguousDNA())\n",
      " |      >>> ambig_dna_seq\n",
      " |      Seq('ACRGT', IUPACAmbiguousDNA())\n",
      " |      \n",
      " |      If we add the ambiguous and unambiguous IUPAC DNA alphabets, we get\n",
      " |      the more general ambiguous IUPAC DNA alphabet:\n",
      " |      \n",
      " |      >>> unamb_dna_seq + ambig_dna_seq\n",
      " |      Seq('ACGTACRGT', IUPACAmbiguousDNA())\n",
      " |      \n",
      " |      However, if the default generic alphabet is included, the result is\n",
      " |      a generic alphabet:\n",
      " |      \n",
      " |      >>> Seq(\"\") + ambig_dna_seq\n",
      " |      Seq('ACRGT', Alphabet())\n",
      " |      \n",
      " |      You can't add RNA and DNA sequences:\n",
      " |      \n",
      " |      >>> from Bio.Alphabet import generic_dna, generic_rna\n",
      " |      >>> Seq(\"ACGT\", generic_dna) + Seq(\"ACGU\", generic_rna)\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      TypeError: Incompatible alphabets DNAAlphabet() and RNAAlphabet()\n",
      " |      \n",
      " |      You can't add nucleotide and protein sequences:\n",
      " |      \n",
      " |      >>> from Bio.Alphabet import generic_dna, generic_protein\n",
      " |      >>> Seq(\"ACGT\", generic_dna) + Seq(\"MELKI\", generic_protein)\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      TypeError: Incompatible alphabets DNAAlphabet() and ProteinAlphabet()\n",
      " |  \n",
      " |  __contains__(self, char)\n",
      " |      Implements the 'in' keyword, like a python string.\n",
      " |      \n",
      " |      e.g.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna, generic_rna, generic_protein\n",
      " |      >>> my_dna = Seq(\"ATATGAAATTTGAAAA\", generic_dna)\n",
      " |      >>> \"AAA\" in my_dna\n",
      " |      True\n",
      " |      >>> Seq(\"AAA\") in my_dna\n",
      " |      True\n",
      " |      >>> Seq(\"AAA\", generic_dna) in my_dna\n",
      " |      True\n",
      " |      \n",
      " |      Like other Seq methods, this will raise a type error if another Seq\n",
      " |      (or Seq like) object with an incompatible alphabet is used:\n",
      " |      \n",
      " |      >>> Seq(\"AAA\", generic_rna) in my_dna\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      TypeError: Incompatible alphabets DNAAlphabet() and RNAAlphabet()\n",
      " |      >>> Seq(\"AAA\", generic_protein) in my_dna\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      TypeError: Incompatible alphabets DNAAlphabet() and ProteinAlphabet()\n",
      " |  \n",
      " |  __eq__(self, other)\n",
      " |      Compare the sequence to another sequence or a string (README).\n",
      " |      \n",
      " |      Historically comparing Seq objects has done Python object comparison.\n",
      " |      After considerable discussion (keeping in mind constraints of the\n",
      " |      Python language, hashes and dictionary support), Biopython now uses\n",
      " |      simple string comparison (with a warning about the change).\n",
      " |      \n",
      " |      Note that incompatible alphabets (e.g. DNA to RNA) will trigger a\n",
      " |      warning.\n",
      " |      \n",
      " |      During this transition period, please just do explicit comparisons:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna\n",
      " |      >>> seq1 = Seq(\"ACGT\")\n",
      " |      >>> seq2 = Seq(\"ACGT\")\n",
      " |      >>> id(seq1) == id(seq2)\n",
      " |      False\n",
      " |      >>> str(seq1) == str(seq2)\n",
      " |      True\n",
      " |      \n",
      " |      The new behaviour is to use string-like equality:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna\n",
      " |      >>> seq1 == seq2\n",
      " |      True\n",
      " |      >>> seq1 == \"ACGT\"\n",
      " |      True\n",
      " |      >>> seq1 == Seq(\"ACGT\", generic_dna)\n",
      " |      True\n",
      " |  \n",
      " |  __getitem__(self, index)\n",
      " |      Returns a subsequence of single letter, use my_seq[index].\n",
      " |  \n",
      " |  __hash__(self)\n",
      " |      Hash for comparison.\n",
      " |      \n",
      " |      See the __cmp__ documentation - this has changed from past\n",
      " |      versions of Biopython!\n",
      " |  \n",
      " |  __init__(self, data, alphabet=Alphabet())\n",
      " |      Create a Seq object.\n",
      " |      \n",
      " |      Arguments:\n",
      " |          - seq - Sequence, required (string)\n",
      " |          - alphabet - Optional argument, an Alphabet object from Bio.Alphabet\n",
      " |      \n",
      " |      You will typically use Bio.SeqIO to read in sequences from files as\n",
      " |      SeqRecord objects, whose sequence will be exposed as a Seq object via\n",
      " |      the seq property.\n",
      " |      \n",
      " |      However, will often want to create your own Seq objects directly:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> my_seq = Seq(\"MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF\",\n",
      " |      ...              IUPAC.protein)\n",
      " |      >>> my_seq\n",
      " |      Seq('MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF', IUPACProtein())\n",
      " |      >>> print(my_seq)\n",
      " |      MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF\n",
      " |      >>> my_seq.alphabet\n",
      " |      IUPACProtein()\n",
      " |  \n",
      " |  __le__(self, other)\n",
      " |      Less than or equal, see __eq__ documentation.\n",
      " |  \n",
      " |  __len__(self)\n",
      " |      Returns the length of the sequence, use len(my_seq).\n",
      " |  \n",
      " |  __lt__(self, other)\n",
      " |      Less than, see __eq__ documentation.\n",
      " |  \n",
      " |  __ne__(self, other)\n",
      " |      Not equal, see __eq__ documentation.\n",
      " |  \n",
      " |  __radd__(self, other)\n",
      " |      Adding a sequence on the left.\n",
      " |      \n",
      " |      If adding a string to a Seq, the alphabet is preserved:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_protein\n",
      " |      >>> \"LV\" + Seq(\"MELKI\", generic_protein)\n",
      " |      Seq('LVMELKI', ProteinAlphabet())\n",
      " |      \n",
      " |      Adding two Seq (like) objects is handled via the __add__ method.\n",
      " |  \n",
      " |  __repr__(self)\n",
      " |      Returns a (truncated) representation of the sequence for debugging.\n",
      " |  \n",
      " |  __str__(self)\n",
      " |      Returns the full sequence as a python string, use str(my_seq).\n",
      " |      \n",
      " |      Note that Biopython 1.44 and earlier would give a truncated\n",
      " |      version of repr(my_seq) for str(my_seq).  If you are writing code\n",
      " |      which need to be backwards compatible with old Biopython, you\n",
      " |      should continue to use my_seq.tostring() rather than str(my_seq).\n",
      " |  \n",
      " |  back_transcribe(self)\n",
      " |      Returns the DNA sequence from an RNA sequence. New Seq object.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> messenger_rna = Seq(\"AUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAG\",\n",
      " |      ...                     IUPAC.unambiguous_rna)\n",
      " |      >>> messenger_rna\n",
      " |      Seq('AUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAG', IUPACUnambiguousRNA())\n",
      " |      >>> messenger_rna.back_transcribe()\n",
      " |      Seq('ATGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG', IUPACUnambiguousDNA())\n",
      " |      \n",
      " |      Trying to back-transcribe a protein or DNA sequence raises an\n",
      " |      exception:\n",
      " |      \n",
      " |      >>> my_protein = Seq(\"MAIVMGR\", IUPAC.protein)\n",
      " |      >>> my_protein.back_transcribe()\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      ValueError: Proteins cannot be back transcribed!\n",
      " |  \n",
      " |  complement(self)\n",
      " |      Returns the complement sequence. New Seq object.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> my_dna = Seq(\"CCCCCGATAG\", IUPAC.unambiguous_dna)\n",
      " |      >>> my_dna\n",
      " |      Seq('CCCCCGATAG', IUPACUnambiguousDNA())\n",
      " |      >>> my_dna.complement()\n",
      " |      Seq('GGGGGCTATC', IUPACUnambiguousDNA())\n",
      " |      \n",
      " |      You can of course used mixed case sequences,\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna\n",
      " |      >>> my_dna = Seq(\"CCCCCgatA-GD\", generic_dna)\n",
      " |      >>> my_dna\n",
      " |      Seq('CCCCCgatA-GD', DNAAlphabet())\n",
      " |      >>> my_dna.complement()\n",
      " |      Seq('GGGGGctaT-CH', DNAAlphabet())\n",
      " |      \n",
      " |      Note in the above example, ambiguous character D denotes\n",
      " |      G, A or T so its complement is H (for C, T or A).\n",
      " |      \n",
      " |      Trying to complement a protein sequence raises an exception.\n",
      " |      \n",
      " |      >>> my_protein = Seq(\"MAIVMGR\", IUPAC.protein)\n",
      " |      >>> my_protein.complement()\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      ValueError: Proteins do not have complements!\n",
      " |  \n",
      " |  count(self, sub, start=0, end=9223372036854775807)\n",
      " |      Non-overlapping count method, like that of a python string.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name,\n",
      " |      which does a non-overlapping count!\n",
      " |      \n",
      " |      Returns an integer, the number of occurrences of substring\n",
      " |      argument sub in the (sub)sequence given by [start:end].\n",
      " |      Optional arguments start and end are interpreted as in slice\n",
      " |      notation.\n",
      " |      \n",
      " |      Arguments:\n",
      " |          - sub - a string or another Seq object to look for\n",
      " |          - start - optional integer, slice start\n",
      " |          - end - optional integer, slice end\n",
      " |      \n",
      " |      e.g.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_seq = Seq(\"AAAATGA\")\n",
      " |      >>> print(my_seq.count(\"A\"))\n",
      " |      5\n",
      " |      >>> print(my_seq.count(\"ATG\"))\n",
      " |      1\n",
      " |      >>> print(my_seq.count(Seq(\"AT\")))\n",
      " |      1\n",
      " |      >>> print(my_seq.count(\"AT\", 2, -1))\n",
      " |      1\n",
      " |      \n",
      " |      HOWEVER, please note because python strings and Seq objects (and\n",
      " |      MutableSeq objects) do a non-overlapping search, this may not give\n",
      " |      the answer you expect:\n",
      " |      \n",
      " |      >>> \"AAAA\".count(\"AA\")\n",
      " |      2\n",
      " |      >>> print(Seq(\"AAAA\").count(\"AA\"))\n",
      " |      2\n",
      " |      \n",
      " |      An overlapping search would give the answer as three!\n",
      " |  \n",
      " |  endswith(self, suffix, start=0, end=9223372036854775807)\n",
      " |      Does the Seq end with the given suffix?  Returns True/False.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Return True if the sequence ends with the specified suffix\n",
      " |      (a string or another Seq object), False otherwise.\n",
      " |      With optional start, test sequence beginning at that position.\n",
      " |      With optional end, stop comparing sequence at that position.\n",
      " |      suffix can also be a tuple of strings to try.  e.g.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_rna = Seq(\"GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG\")\n",
      " |      >>> my_rna.endswith(\"UUG\")\n",
      " |      True\n",
      " |      >>> my_rna.endswith(\"AUG\")\n",
      " |      False\n",
      " |      >>> my_rna.endswith(\"AUG\", 0, 18)\n",
      " |      True\n",
      " |      >>> my_rna.endswith((\"UCC\", \"UCA\", \"UUG\"))\n",
      " |      True\n",
      " |  \n",
      " |  find(self, sub, start=0, end=9223372036854775807)\n",
      " |      Find method, like that of a python string.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Returns an integer, the index of the first occurrence of substring\n",
      " |      argument sub in the (sub)sequence given by [start:end].\n",
      " |      \n",
      " |      Arguments:\n",
      " |          - sub - a string or another Seq object to look for\n",
      " |          - start - optional integer, slice start\n",
      " |          - end - optional integer, slice end\n",
      " |      \n",
      " |      Returns -1 if the subsequence is NOT found.\n",
      " |      \n",
      " |      e.g. Locating the first typical start codon, AUG, in an RNA sequence:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_rna = Seq(\"GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG\")\n",
      " |      >>> my_rna.find(\"AUG\")\n",
      " |      3\n",
      " |  \n",
      " |  lower(self)\n",
      " |      Returns a lower case copy of the sequence.\n",
      " |      \n",
      " |      This will adjust the alphabet if required. Note that the IUPAC alphabets\n",
      " |      are upper case only, and thus a generic alphabet must be substituted.\n",
      " |      \n",
      " |      >>> from Bio.Alphabet import Gapped, generic_dna\n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_seq = Seq(\"CGGTACGCTTATGTCACGTAG*AAAAAA\", Gapped(IUPAC.unambiguous_dna, \"*\"))\n",
      " |      >>> my_seq\n",
      " |      Seq('CGGTACGCTTATGTCACGTAG*AAAAAA', Gapped(IUPACUnambiguousDNA(), '*'))\n",
      " |      >>> my_seq.lower()\n",
      " |      Seq('cggtacgcttatgtcacgtag*aaaaaa', Gapped(DNAAlphabet(), '*'))\n",
      " |      \n",
      " |      See also the upper method.\n",
      " |  \n",
      " |  lstrip(self, chars=None)\n",
      " |      Returns a new Seq object with leading (left) end stripped.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Optional argument chars defines which characters to remove.  If\n",
      " |      omitted or None (default) then as for the python string method,\n",
      " |      this defaults to removing any white space.\n",
      " |      \n",
      " |      e.g. print(my_seq.lstrip(\"-\"))\n",
      " |      \n",
      " |      See also the strip and rstrip methods.\n",
      " |  \n",
      " |  reverse_complement(self)\n",
      " |      Returns the reverse complement sequence. New Seq object.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> my_dna = Seq(\"CCCCCGATAGNR\", IUPAC.ambiguous_dna)\n",
      " |      >>> my_dna\n",
      " |      Seq('CCCCCGATAGNR', IUPACAmbiguousDNA())\n",
      " |      >>> my_dna.reverse_complement()\n",
      " |      Seq('YNCTATCGGGGG', IUPACAmbiguousDNA())\n",
      " |      \n",
      " |      Note in the above example, since R = G or A, its complement\n",
      " |      is Y (which denotes C or T).\n",
      " |      \n",
      " |      You can of course used mixed case sequences,\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna\n",
      " |      >>> my_dna = Seq(\"CCCCCgatA-G\", generic_dna)\n",
      " |      >>> my_dna\n",
      " |      Seq('CCCCCgatA-G', DNAAlphabet())\n",
      " |      >>> my_dna.reverse_complement()\n",
      " |      Seq('C-TatcGGGGG', DNAAlphabet())\n",
      " |      \n",
      " |      Trying to complement a protein sequence raises an exception:\n",
      " |      \n",
      " |      >>> my_protein = Seq(\"MAIVMGR\", IUPAC.protein)\n",
      " |      >>> my_protein.reverse_complement()\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      ValueError: Proteins do not have complements!\n",
      " |  \n",
      " |  rfind(self, sub, start=0, end=9223372036854775807)\n",
      " |      Find from right method, like that of a python string.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Returns an integer, the index of the last (right most) occurrence of\n",
      " |      substring argument sub in the (sub)sequence given by [start:end].\n",
      " |      \n",
      " |      Arguments:\n",
      " |          - sub - a string or another Seq object to look for\n",
      " |          - start - optional integer, slice start\n",
      " |          - end - optional integer, slice end\n",
      " |      \n",
      " |      Returns -1 if the subsequence is NOT found.\n",
      " |      \n",
      " |      e.g. Locating the last typical start codon, AUG, in an RNA sequence:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_rna = Seq(\"GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG\")\n",
      " |      >>> my_rna.rfind(\"AUG\")\n",
      " |      15\n",
      " |  \n",
      " |  rsplit(self, sep=None, maxsplit=-1)\n",
      " |      Right split method, like that of a python string.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Return a list of the 'words' in the string (as Seq objects),\n",
      " |      using sep as the delimiter string.  If maxsplit is given, at\n",
      " |      most maxsplit splits are done COUNTING FROM THE RIGHT.\n",
      " |      If maxsplit is omitted, all splits are made.\n",
      " |      \n",
      " |      Following the python string method, sep will by default be any\n",
      " |      white space (tabs, spaces, newlines) but this is unlikely to\n",
      " |      apply to biological sequences.\n",
      " |      \n",
      " |      e.g. print(my_seq.rsplit(\"*\",1))\n",
      " |      \n",
      " |      See also the split method.\n",
      " |  \n",
      " |  rstrip(self, chars=None)\n",
      " |      Returns a new Seq object with trailing (right) end stripped.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Optional argument chars defines which characters to remove.  If\n",
      " |      omitted or None (default) then as for the python string method,\n",
      " |      this defaults to removing any white space.\n",
      " |      \n",
      " |      e.g. Removing a nucleotide sequence's polyadenylation (poly-A tail):\n",
      " |      \n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_seq = Seq(\"CGGTACGCTTATGTCACGTAGAAAAAA\", IUPAC.unambiguous_dna)\n",
      " |      >>> my_seq\n",
      " |      Seq('CGGTACGCTTATGTCACGTAGAAAAAA', IUPACUnambiguousDNA())\n",
      " |      >>> my_seq.rstrip(\"A\")\n",
      " |      Seq('CGGTACGCTTATGTCACGTAG', IUPACUnambiguousDNA())\n",
      " |      \n",
      " |      See also the strip and lstrip methods.\n",
      " |  \n",
      " |  split(self, sep=None, maxsplit=-1)\n",
      " |      Split method, like that of a python string.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Return a list of the 'words' in the string (as Seq objects),\n",
      " |      using sep as the delimiter string.  If maxsplit is given, at\n",
      " |      most maxsplit splits are done.  If maxsplit is omitted, all\n",
      " |      splits are made.\n",
      " |      \n",
      " |      Following the python string method, sep will by default be any\n",
      " |      white space (tabs, spaces, newlines) but this is unlikely to\n",
      " |      apply to biological sequences.\n",
      " |      \n",
      " |      e.g.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_rna = Seq(\"GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG\")\n",
      " |      >>> my_aa = my_rna.translate()\n",
      " |      >>> my_aa\n",
      " |      Seq('VMAIVMGR*KGAR*L', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |      >>> my_aa.split(\"*\")\n",
      " |      [Seq('VMAIVMGR', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('KGAR', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('L', HasStopCodon(ExtendedIUPACProtein(), '*'))]\n",
      " |      >>> my_aa.split(\"*\", 1)\n",
      " |      [Seq('VMAIVMGR', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('KGAR*L', HasStopCodon(ExtendedIUPACProtein(), '*'))]\n",
      " |      \n",
      " |      See also the rsplit method:\n",
      " |      \n",
      " |      >>> my_aa.rsplit(\"*\", 1)\n",
      " |      [Seq('VMAIVMGR*KGAR', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('L', HasStopCodon(ExtendedIUPACProtein(), '*'))]\n",
      " |  \n",
      " |  startswith(self, prefix, start=0, end=9223372036854775807)\n",
      " |      Does the Seq start with the given prefix?  Returns True/False.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Return True if the sequence starts with the specified prefix\n",
      " |      (a string or another Seq object), False otherwise.\n",
      " |      With optional start, test sequence beginning at that position.\n",
      " |      With optional end, stop comparing sequence at that position.\n",
      " |      prefix can also be a tuple of strings to try.  e.g.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_rna = Seq(\"GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG\")\n",
      " |      >>> my_rna.startswith(\"GUC\")\n",
      " |      True\n",
      " |      >>> my_rna.startswith(\"AUG\")\n",
      " |      False\n",
      " |      >>> my_rna.startswith(\"AUG\", 3)\n",
      " |      True\n",
      " |      >>> my_rna.startswith((\"UCC\", \"UCA\", \"UCG\"), 1)\n",
      " |      True\n",
      " |  \n",
      " |  strip(self, chars=None)\n",
      " |      Returns a new Seq object with leading and trailing ends stripped.\n",
      " |      \n",
      " |      This behaves like the python string method of the same name.\n",
      " |      \n",
      " |      Optional argument chars defines which characters to remove.  If\n",
      " |      omitted or None (default) then as for the python string method,\n",
      " |      this defaults to removing any white space.\n",
      " |      \n",
      " |      e.g. print(my_seq.strip(\"-\"))\n",
      " |      \n",
      " |      See also the lstrip and rstrip methods.\n",
      " |  \n",
      " |  tomutable(self)\n",
      " |      Returns the full sequence as a MutableSeq object.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> my_seq = Seq(\"MKQHKAMIVALIVICITAVVAAL\",\n",
      " |      ...              IUPAC.protein)\n",
      " |      >>> my_seq\n",
      " |      Seq('MKQHKAMIVALIVICITAVVAAL', IUPACProtein())\n",
      " |      >>> my_seq.tomutable()\n",
      " |      MutableSeq('MKQHKAMIVALIVICITAVVAAL', IUPACProtein())\n",
      " |      \n",
      " |      Note that the alphabet is preserved.\n",
      " |  \n",
      " |  tostring(self)\n",
      " |      Returns the full sequence as a python string (DEPRECATED).\n",
      " |      \n",
      " |      You are now encouraged to use str(my_seq) instead of\n",
      " |      my_seq.tostring().\n",
      " |  \n",
      " |  transcribe(self)\n",
      " |      Returns the RNA sequence from a DNA sequence. New Seq object.\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC\n",
      " |      >>> coding_dna = Seq(\"ATGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG\",\n",
      " |      ...                  IUPAC.unambiguous_dna)\n",
      " |      >>> coding_dna\n",
      " |      Seq('ATGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG', IUPACUnambiguousDNA())\n",
      " |      >>> coding_dna.transcribe()\n",
      " |      Seq('AUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAG', IUPACUnambiguousRNA())\n",
      " |      \n",
      " |      Trying to transcribe a protein or RNA sequence raises an exception:\n",
      " |      \n",
      " |      >>> my_protein = Seq(\"MAIVMGR\", IUPAC.protein)\n",
      " |      >>> my_protein.transcribe()\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      ValueError: Proteins cannot be transcribed!\n",
      " |  \n",
      " |  translate(self, table='Standard', stop_symbol='*', to_stop=False, cds=False)\n",
      " |      Turns a nucleotide sequence into a protein sequence. New Seq object.\n",
      " |      \n",
      " |      This method will translate DNA or RNA sequences, and those with a\n",
      " |      nucleotide or generic alphabet.  Trying to translate a protein\n",
      " |      sequence raises an exception.\n",
      " |      \n",
      " |      Arguments:\n",
      " |          - table - Which codon table to use?  This can be either a name\n",
      " |            (string), an NCBI identifier (integer), or a CodonTable\n",
      " |            object (useful for non-standard genetic codes).  This\n",
      " |            defaults to the \"Standard\" table.\n",
      " |          - stop_symbol - Single character string, what to use for terminators.\n",
      " |            This defaults to the asterisk, \"*\".\n",
      " |          - to_stop - Boolean, defaults to False meaning do a full translation\n",
      " |            continuing on past any stop codons (translated as the\n",
      " |            specified stop_symbol).  If True, translation is\n",
      " |            terminated at the first in frame stop codon (and the\n",
      " |            stop_symbol is not appended to the returned protein\n",
      " |            sequence).\n",
      " |          - cds - Boolean, indicates this is a complete CDS.  If True,\n",
      " |            this checks the sequence starts with a valid alternative start\n",
      " |            codon (which will be translated as methionine, M), that the\n",
      " |            sequence length is a multiple of three, and that there is a\n",
      " |            single in frame stop codon at the end (this will be excluded\n",
      " |            from the protein sequence, regardless of the to_stop option).\n",
      " |            If these tests fail, an exception is raised.\n",
      " |      \n",
      " |      e.g. Using the standard table:\n",
      " |      \n",
      " |      >>> coding_dna = Seq(\"GTGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG\")\n",
      " |      >>> coding_dna.translate()\n",
      " |      Seq('VAIVMGR*KGAR*', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |      >>> coding_dna.translate(stop_symbol=\"@\")\n",
      " |      Seq('VAIVMGR@KGAR@', HasStopCodon(ExtendedIUPACProtein(), '@'))\n",
      " |      >>> coding_dna.translate(to_stop=True)\n",
      " |      Seq('VAIVMGR', ExtendedIUPACProtein())\n",
      " |      \n",
      " |      Now using NCBI table 2, where TGA is not a stop codon:\n",
      " |      \n",
      " |      >>> coding_dna.translate(table=2)\n",
      " |      Seq('VAIVMGRWKGAR*', HasStopCodon(ExtendedIUPACProtein(), '*'))\n",
      " |      >>> coding_dna.translate(table=2, to_stop=True)\n",
      " |      Seq('VAIVMGRWKGAR', ExtendedIUPACProtein())\n",
      " |      \n",
      " |      In fact, GTG is an alternative start codon under NCBI table 2, meaning\n",
      " |      this sequence could be a complete CDS:\n",
      " |      \n",
      " |      >>> coding_dna.translate(table=2, cds=True)\n",
      " |      Seq('MAIVMGRWKGAR', ExtendedIUPACProtein())\n",
      " |      \n",
      " |      It isn't a valid CDS under NCBI table 1, due to both the start codon and\n",
      " |      also the in frame stop codons:\n",
      " |      \n",
      " |      >>> coding_dna.translate(table=1, cds=True)\n",
      " |      Traceback (most recent call last):\n",
      " |          ...\n",
      " |      TranslationError: First codon 'GTG' is not a start codon\n",
      " |      \n",
      " |      If the sequence has no in-frame stop codon, then the to_stop argument\n",
      " |      has no effect:\n",
      " |      \n",
      " |      >>> coding_dna2 = Seq(\"TTGGCCATTGTAATGGGCCGC\")\n",
      " |      >>> coding_dna2.translate()\n",
      " |      Seq('LAIVMGR', ExtendedIUPACProtein())\n",
      " |      >>> coding_dna2.translate(to_stop=True)\n",
      " |      Seq('LAIVMGR', ExtendedIUPACProtein())\n",
      " |      \n",
      " |      NOTE - Ambiguous codons like \"TAN\" or \"NNN\" could be an amino acid\n",
      " |      or a stop codon.  These are translated as \"X\".  Any invalid codon\n",
      " |      (e.g. \"TA?\" or \"T-A\") will throw a TranslationError.\n",
      " |      \n",
      " |      NOTE - Does NOT support gapped sequences.\n",
      " |      \n",
      " |      NOTE - This does NOT behave like the python string's translate\n",
      " |      method.  For that use str(my_seq).translate(...) instead.\n",
      " |  \n",
      " |  ungap(self, gap=None)\n",
      " |      Return a copy of the sequence without the gap character(s).\n",
      " |      \n",
      " |      The gap character can be specified in two ways - either as an explicit\n",
      " |      argument, or via the sequence's alphabet. For example:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna\n",
      " |      >>> my_dna = Seq(\"-ATA--TGAAAT-TTGAAAA\", generic_dna)\n",
      " |      >>> my_dna\n",
      " |      Seq('-ATA--TGAAAT-TTGAAAA', DNAAlphabet())\n",
      " |      >>> my_dna.ungap(\"-\")\n",
      " |      Seq('ATATGAAATTTGAAAA', DNAAlphabet())\n",
      " |      \n",
      " |      If the gap character is not given as an argument, it will be taken from\n",
      " |      the sequence's alphabet (if defined). Notice that the returned sequence's\n",
      " |      alphabet is adjusted since it no longer requires a gapped alphabet:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC, Gapped, HasStopCodon\n",
      " |      >>> my_pro = Seq(\"MVVLE=AD*\", HasStopCodon(Gapped(IUPAC.protein, \"=\")))\n",
      " |      >>> my_pro\n",
      " |      Seq('MVVLE=AD*', HasStopCodon(Gapped(IUPACProtein(), '='), '*'))\n",
      " |      >>> my_pro.ungap()\n",
      " |      Seq('MVVLEAD*', HasStopCodon(IUPACProtein(), '*'))\n",
      " |      \n",
      " |      Or, with a simpler gapped DNA example:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import IUPAC, Gapped\n",
      " |      >>> my_seq = Seq(\"CGGGTAG=AAAAAA\", Gapped(IUPAC.unambiguous_dna, \"=\"))\n",
      " |      >>> my_seq\n",
      " |      Seq('CGGGTAG=AAAAAA', Gapped(IUPACUnambiguousDNA(), '='))\n",
      " |      >>> my_seq.ungap()\n",
      " |      Seq('CGGGTAGAAAAAA', IUPACUnambiguousDNA())\n",
      " |      \n",
      " |      As long as it is consistent with the alphabet, although it is redundant,\n",
      " |      you can still supply the gap character as an argument to this method:\n",
      " |      \n",
      " |      >>> my_seq\n",
      " |      Seq('CGGGTAG=AAAAAA', Gapped(IUPACUnambiguousDNA(), '='))\n",
      " |      >>> my_seq.ungap(\"=\")\n",
      " |      Seq('CGGGTAGAAAAAA', IUPACUnambiguousDNA())\n",
      " |      \n",
      " |      However, if the gap character given as the argument disagrees with that\n",
      " |      declared in the alphabet, an exception is raised:\n",
      " |      \n",
      " |      >>> my_seq\n",
      " |      Seq('CGGGTAG=AAAAAA', Gapped(IUPACUnambiguousDNA(), '='))\n",
      " |      >>> my_seq.ungap(\"-\")\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      ValueError: Gap '-' does not match '=' from alphabet\n",
      " |      \n",
      " |      Finally, if a gap character is not supplied, and the alphabet does not\n",
      " |      define one, an exception is raised:\n",
      " |      \n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> from Bio.Alphabet import generic_dna\n",
      " |      >>> my_dna = Seq(\"ATA--TGAAAT-TTGAAAA\", generic_dna)\n",
      " |      >>> my_dna\n",
      " |      Seq('ATA--TGAAAT-TTGAAAA', DNAAlphabet())\n",
      " |      >>> my_dna.ungap()\n",
      " |      Traceback (most recent call last):\n",
      " |         ...\n",
      " |      ValueError: Gap character not given and not defined in alphabet\n",
      " |  \n",
      " |  upper(self)\n",
      " |      Returns an upper case copy of the sequence.\n",
      " |      \n",
      " |      >>> from Bio.Alphabet import HasStopCodon, generic_protein\n",
      " |      >>> from Bio.Seq import Seq\n",
      " |      >>> my_seq = Seq(\"VHLTPeeK*\", HasStopCodon(generic_protein))\n",
      " |      >>> my_seq\n",
      " |      Seq('VHLTPeeK*', HasStopCodon(ProteinAlphabet(), '*'))\n",
      " |      >>> my_seq.lower()\n",
      " |      Seq('vhltpeek*', HasStopCodon(ProteinAlphabet(), '*'))\n",
      " |      >>> my_seq.upper()\n",
      " |      Seq('VHLTPEEK*', HasStopCodon(ProteinAlphabet(), '*'))\n",
      " |      \n",
      " |      This will adjust the alphabet if required. See also the lower method.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data descriptors defined here:\n",
      " |  \n",
      " |  __dict__\n",
      " |      dictionary for instance variables (if defined)\n",
      " |  \n",
      " |  __weakref__\n",
      " |      list of weak references to the object (if defined)\n",
      "\n",
      "None\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "print (help(Seq))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Alphabet()\n",
      "['__add__', '__class__', '__contains__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getitem__', '__gt__', '__hash__', '__init__', '__le__', '__len__', '__lt__', '__module__', '__ne__', '__new__', '__radd__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_data', '_get_seq_str_and_check_alphabet', 'alphabet', 'back_transcribe', 'complement', 'count', 'endswith', 'find', 'lower', 'lstrip', 'reverse_complement', 'rfind', 'rsplit', 'rstrip', 'split', 'startswith', 'strip', 'tomutable', 'tostring', 'transcribe', 'translate', 'ungap', 'upper']\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio.Alphabet import IUPAC\n",
    "prot = Seq(\"kljsdalkjfa\", IUPAC.protein)\n",
    "mydna = Seq(\"catatgatgatg\")\n",
    "print(mydna.alphabet)\n",
    "print(dir(prot))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "original catatgatgatg\n",
      "reverse complement catcatcatatg\n",
      "complement gtatactactac\n"
     ]
    }
   ],
   "source": [
    "print(\"original\", mydna)\n",
    "print(\"reverse complement\", mydna.reverse_complement())\n",
    "print(\"complement\",  mydna.complement())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "raw_sequence = \"\"\"\n",
    "TTGGCCACCACTGCCTTTGTGCGCGATCTCGCGCTGCCATTGGCTAACTCGGGAAAGTGGGAAGCGTGAA\n",
    "GGAGGGACCCTGAGGTAGAGGGTCAGGGGTTAGTGAGGCCGGAAGTGAGTGTAATAAAGTTTCTCCAGGG\n",
    "AGGCAGGGCCCGGGGAGAAAGTTGGAGCGGTAACCTAAGCTGGCAGTGGCGTGATCCGGCACCAAATCGG\n",
    "CCCGCGGTGCGGTGCGGAGACTCCATGAGGCCCTGGTGAGTCTTGACTTTTTTCTCCTTAGGTGACTCAC\n",
    "TCTTTCACCCCGGAACCCCTCTCCCGGGACCCCACTTCCGGTCTTCATTTATAAGTCCCGCCCCTGAGAC\n",
    "CCCTCTCCCCGGAAGTCCCACCTCCCAACTTCAGGGGGCTCCCCCGAACCCCGACTTGATTTTTATTTTT\n",
    "TGTCTCCGACCCCACCCTGTCGCCCAGTTTTTCTGGAAGTCCCCACCCCCCGACACGTTCCAAGCCCTAC\n",
    "TACGAGAAACCCTTCGTTACTCCTTCCATTCTCCGAAACTCCGCCTCCTAGCAACGATCCTCGGCTAACT\n",
    "CCTGTCAAAAGGCCTATCTCCCAAAAGGGCAAAACCCTGCTCTTCAAAGAACGATTCGAATTTGTTGAAG\n",
    "CTCCGCCCCTCCGAAGCCACGCCCCCTCCGTAATCCACGTCTGACTGTCCAGAAAGCCTCCCCGCCCAGA\n",
    "AGTCTGAAGTTCGCGTCCTTAGGGGACCACGCCCCCTTAGGAGCTTTATCTTTGTGGGACCTGGGTCATC\n",
    "AGCTGAAAAGCCTTGTCGTCAATCCCTGAAACTCCACCTGCTTATATTTTCCTGTCCTTTCTCTTTTTGG\n",
    "GCCTTGGGATCAGTTTTCCGAGCACTCTCGTGCTCCCCCAGGTGGGAAGTCAAAGTTCCCATCTCTGGCT\n",
    "GTAAGGAGTGAGATTAAAGGCAGCCGTTTTCCGTTTTCCTTTCTATCTTTTTTTTTTTTTTTTTTTTTTT\n",
    "TAAGACGGAGTTTCGCTCTTATCGTCCAGGCTGGAGTGCAATGGTGCGATCTCGGCTCACTGCAACCTCC\n",
    "GCCTCCTGAGTTCAAGTGATTCTCCTGCCTCAGCCTCCTGAGTAGCTGGGATTACAGGCGCCCACCACCA\n",
    "CGCCCGGCTAATTTTTGTATTTTTAGTAGAGACGGGGTTTCCCCATGTTGGCTAGGCTAGCCTCGAACTC\n",
    "CTGACCTCAGGTGCTCCACCCTTATCAGCCTCCTTAAGTGTTAGGATTACAGGCGTTAGCCACTGCGCCC\n",
    "GTCCCCGTTTTCCTCTTTCTTAACCTCCTCCAAGGGTAGAACTATTTGAAAGGGCTGAACGTGGGGTTTG\n",
    "GCACGAGAGGGAGAGAGAGACTTGCCCTCAGACTCCCAGCCTCACTTCCTGGCTTTCCCTTCCCAGGACC\n",
    "AACCTCATCCCAGAAGTAGCTCTAGGCCTTCCTAGACATAAGTGGATAAGACCATGGATTTGTTGGTGTG\n",
    "TGGCATGTGGGTGTGACATGAGCAAAAGTTTTCTACTACCAGAAGAACGGTAGTTTGGAGTGGGGGTAGA\n",
    "TAATAGTGCTCTTTTTGTTGTTTGATTTTTTTTTTTTTAAGGGACAGTTCAGTCGCTACTTCTCTGTCTC\n",
    "TTCCCCTCCCCCTAACCAGAGTTCTTGTCTGGGCATTCCTCCTGGCCCCGTTTGCCCAGCCTTCCCTGCT\n",
    "GGAAACATTCCTGAGGCTTTCGCCATTTCCTGCCACTTCCCGCCCCTGTCCCCTCCTCCCAGTTGGGCTG\n",
    "CGAAGCCACACACAGCTTTTCCCCTACACCCAATCACCTGGACTTCCAGTAGGCAATCTCCTTTCTCCCT\n",
    "ATGTAAATTTCAAGAGTCATACGTCTTTCTTTTCTTCCGAACGCCCACACCCGTTGGTCCTTCCCTGGCT\n",
    "CTGGATACTGTAGGCTCCCACAGAACATCATCTCTACAGATGAGGGACATCCTAGTGTGTGTTTGTGTGT\n",
    "TTTGGGGGTGTGAGGCTATGGTGGAATCCTTTCCTCATTCTCTCTATGCCCGAGAGCTGGCTGGAAATGA\n",
    "AGTGATAAAATTCCTTCTCAGGATTGGCTGGCAGAGTGACAAAGCTCTTCTGGAATGAGCACTAAGGGGA\n",
    "GATGTAAGAGTTAAGCACAGCATCTAACACACTTCAAGTGTTAATGTTGTTAATCAGGATAATTATCTAT\n",
    "CTATATTTTAATTCTCTGAAGTAAATCTAAGGGTAATTCCAAGGTGAGCTGGGGGGAATACGTATTTTTT\n",
    "TTTTTTTGAGACGGAGTTTTGCTCTTGTCACCTAGGCTGGAGTGCAATGGCTCGATCTCGGCTCACTGCA\n",
    "ACCTCCGCCTCCTGGGTTCGAGCGATTCTCCTGCCTCAGCCTCCTGAGCTGGGATTACAGGTGCCCGCCA\n",
    "CCACGCCCAGCTAATTTTTGTATTTTTAGTAGAGACAGGGTTTCACCATGTTGACCAGGCTGGTCTCGAA\n",
    "CTCCTGACTTCAGGTGATCCACCCGCCTCAGCCTCCCAAAGTGCTGGTATCACAGGTGTGAGCCACCGCG\n",
    "CCCGGTAGGAATACGTATCTTTGTTTTTTTTTTTTTGGTCACTTTCCAGCTTTGCCTGTGGCATTCTCCG\n",
    "GAAGAGCAGTGGCAGGAAGCCGTGGGTGGCAGGAATGGGTTTGGGACTGTTCCCTGTGTCCAGCTTCAAC\n",
    "CCATCTTCTCCTGACACCCCTTTCTCCTGTACTTATTTTTCTCCTGTCCTTATTGACGCAGATGTGTCTT\n",
    "CTGATCTCTCTGTCCCTTCCTGGCCTGTAGTTTCTACTTCTCTTGGAGAACTGTTTCTTCCACCTCCTCC\n",
    "AGCCAGGCTCCTTCCCTGGTGTGAAGGATGCTGTGCTCTTTGGTGGGGCTTAGGGCCCTGTAAGAGGAGG\n",
    "GATGTGAATTGGGGTGGTTTGGAGTGGGAGTGGGGACCTGATTGCCTCATTCTCACCCTTGACAGTGTGC\n",
    "TGTGTTAACAAGCCCTGTGGAAAGGGGAGACAGGACAGTGCTTGGCTATGAGCCATCGGTCTTGGTGTGG\n",
    "TAGATGGCATACGTAGAGAACAAGCCGTGACCAGCATGTCCTCTTGTTCCCTTCTTGCCTGGTAGCTTCC\n",
    "TGATGTATCATCCAGGTTGAACTTTATTCCCCTTTTTTCACCAGCTTTGCTCTTCCTCTTGGCCCTGCTC\n",
    "TGCATCCATTGGGACCACTTACTTCCCATCCCCACTTCGCTAGCTCCAGTCCACAGCTGCTGTAGCAGAG\n",
    "GATGCTAATGGCCCAGGCTTACTGTATGGGGTAGCGGTTGGGGGTGGCCTGGGCTGGGGCCAGCCAGCTT\n",
    "TCGGGCATAGAGAGGGTGGGGGAATTCGAGTGTCCAACCAGGACAGGAAGGGAAATGCTGAGCTCTGAGA\n",
    "TCTCCTCCCTTTTCCACCCCCAGCTGCCTCTGTTTCCTCCCTCATTCCCCTGTACTCCTGCCTGGCTTCC\n",
    "TGCCCCCTTTGGCCTGCGTGTGTCCGGTTTTGAGTGACCCAGACCTCCTATCTCCTTCCTTTCCGTCCCA\n",
    "CTCTTCCATCCTTCCCTCCCACCCCCACTTTACTTCTCCCCAGATTACTAGTTGTTTGTATAGGGCAGGG\n",
    "GAGAGGGGCTTGGATAGCCCTACCCCCTGCTCCAGTTCGAGGTTGGGGCGGGGTGGGGGCTCAGCCCCTC\n",
    "TGAGCCTGTGAGTCAGCACTGTCCTCACGATTGGTCTCTCCCCTTAGCTCCCCGCCCCTGGGGAGGAGCC\n",
    "AGAAGCGCTGGGTCCAGCCTGTTCTCTTCTCAGCCAGAGAAGAGGCTCCACGTTGGGCGGGGATGGGGCC\n",
    "TGAAACTGTCTGGGTCTGAGCTGGGGAGCGGAAGCCACTTGTCCCTCTCCCTCCCCAGGACTTCTGTGAC\n",
    "TCCTGGGCCACAGAGGTCCAACCAGGCTAAGGGCCTGGGGATACCCCCTGCCTGGCCCCCTTGCCCAAAC\n",
    "TGGCAGGGGGGCCAGGCTGGGCAGCAGCCCCTCTTTCACCTCAACTATGGATCTCCTGCCCCCCAAGCCC\n",
    "AAGTACAATCCACTCCGGAATGAGTCTCTGTCATCGCTGGAGGAAGGGGCTTCTGGGTCCACCCCCCCGG\n",
    "AGGAGCTGCCTTCCCCATCAGCTTCATCCCTGGGGCCCATCCTGCCTCCTCTGCCTGGGGACGATAGTCC\n",
    "CACTACCCTGTGCTCCTTCTTCCCCCGGATGAGCAACCTGAGGCTGGCCAACCCGGCTGGGGGGCGCCCA\n",
    "GGGTCTAAGGGGGAGCCAGGAAGGGCAGCTGATGATGGGGAGGGGATCGTAGGGGCAGCCATGCCAGACT\n",
    "CAGGCCCCCTACCCCTCCTCCAGGACATGAACAAGCTGAGTGGAGGCGGCGGGCGCAGGACTCGGGTGGA\n",
    "AGGGGGCCAGCTTGGGGGCGAGGAGTGGACCCGCCACGGGAGCTTTGTCAATAAGCCCACGCGGGGCTGG\n",
    "CTGCATCCCAACGACAAAGTCATGGGACCCGGGGTTTCCTACTTGGTTCGGGTGAGTGAACATCCTCCTC\n",
    "TCCATTCCCCCGCAGACCCCCTCCAGTTCTAATGCTCACTCCAGCTTGGCCGGAATCTCTTTTTTCCTGC\n",
    "TGTTCCCCAACAACCCCGTTTCAGTGTCCTATTTCCTAAACCTTTCTCATCCCTTCCCCGATTCTGTAGC\n",
    "CCCTTCCCTTCCAGCTTTGCCTGGGCCCCCTTCCTTCTTTTCACTCCCCCACCCCCACCCTGGCATGCTT\n",
    "AGCACAATGCCCAGCATGGCAGAAGCTCTTAATAAGTGTTGGAATGAATGACTTCTTGATCTTTCCTCCA\n",
    "TTCTCCTCCCGCTTTCTGGAGTTGTCACTCCATCCCAGCCAATGCCTGTCGGTCCCCTCCAAACTCTCTC\n",
    "CCCAGTGTGTGAGGTCCTTCTAGCCTGCCTTTTCTTTTTCTCCTCATTCCCTGTTGTCTTTCACCTTGGT\n",
    "TGTGGGATGTAGCCACGGGAAGTGGGCAGAACTTCTTCCTCTTTACTTCCCTTAGGGACCATATGGGTAA\n",
    "AGTCCTAGGGCCCTTTGTGAAGGGGCTGGGGTGGGGGAGCCGCTGACCTGCTGTCCTCTTCTCCCTCCCC\n",
    "TTAGTACATGGGTTGTGTGGAGGTCCTCCAGTCAATGCGTGCCCTGGACTTCAACACCCGGACTCAGGTC\n",
    "ACCAGGTTAGTGGGGAGGGAGTGGATTGTGGGAGTCCTGGGATTAGTGAGGCCATGGGAAACAGAGGAGG\n",
    "GGAGTGGGTCACTTGGGTGCTGCTGCTGGGGGCTGATCTTTGCGTAGTAAAGGGGATTGAGAGAAAGCTG\n",
    "AGGTAGTGATTCCGCCTCTTACTGAAAGGAGGCCCCTTTGGCCCGGATCCTCCTTACCTATCCAGCCTTG\n",
    "GGAGTTGTGGGGAAGGTGGCAGAAGCAGCTGGCAGGTGGAGCTGGAGTAGAAAAGGGGGGTAGATGGAAG\n",
    "GTTGCTGCTGCCTTTGGGAAGGGAATGGGACATTTGGTAGCTGGAAGGGGGTGGGGCTATACCCATTGAG\n",
    "GCCCTAACCCAATCAGCCAGGAAGTGGCCTCTCAGTGTCATCAAATATTAAAGGCCCTTAATTCAATCCA\n",
    "CCATGACAAAGAGCCTGGAGTGCCCTGGGAGTCTGGCCTGGCTTTCCCCTCCCCCAGCCCTGTGGCAGCC\n",
    "CCAGTTGAGTGGGAGGCAGGCAGGCAGGCGCTGGGTCTGAGTACCCCTCTTTCCCCAGGGAGGCCATCAG\n",
    "TCTGGTGTGTGAGGCTGTGCCGGGTGCTAAGGGGGCGACAAGGAGGAGAAAGGTACTTGGGGTCCTTGGG\n",
    "GATAGGGATGCTGCTGGGAAATGTGGAGGGGCAGTGGGGAGGCAAGTGTGGGGCATGTGGGGAGAGGGCC\n",
    "AGGAGGCAAAAGTGTGGTTTGCCTGTGCTGGGAACCCTCACCTTCTGAGACCCTGGGCCCTACCTTAGCC\n",
    "CTGTAGCCGCCCGCTCAGCTCTATCCTGGGGAGGAGTAACCTGAAATTTGCTGGAATGCCAATCACTCTC\n",
    "ACCGTCTCCACCAGCAGCCTCAACCTCATGGCCGCAGACTGCAAACAGGTTGGTGAAGGTGGGCAGACGG\n",
    "ACCAGAAACACTGATGGGGGTGCTAAGGGGAGATCTAGTTAGGAGGCCAAAATCTAGGAAGAGGGCTTGA\n",
    "GAGCAGGAGACCAGTAGGGAAGGAGGAAATGAGGATATGGGATAAGGAACCCAGAGAATGAGACTGGGGC\n",
    "ACCAGGGTTCTGGGCCCTGAGGCTGCCTGACACATGGCATTCCCTTCCTTCTTCATCCCCTGCCCTAATT\n",
    "CTCAGATCATCGCCAACCACCACATGCAATCTATCTCATTTGCATCCGGCGGGGATCCGGTGAGTTGGGG\n",
    "AACAGAGCAAAGGCGGTGGGAGAAGAGCAAGGGGCTAGGGATTGGGAAGACATTGATTAGAGAAGTGGGG\n",
    "GTGCTAGGATTGTGGGGATATCTGAAAAGGACTGGGTTTTAGGGGTCAGGAGTAAAGTAGAACTGTAAAA\n",
    "TTTTTTGACCTTCCTCCCCAGGACACAGCCGAGTATGTCGCCTATGTTGCCAAAGACCCTGTGAATCAGA\n",
    "GAGGTGAGGCATGGTGGGGAGCACTGGGTGGGTCTGGCTGTTTGAGGAGGTGAGTAGGACCTTTGGTCAT\n",
    "CTGTACCCACTCTAAGTAGATCTCAGTGGGGGTTCCTGCAGAGGAAGATGAGGGCCCCTGTCTGTGCCTC\n",
    "CTCCCAGTGGGCTCAGCTCACTGACAGCCTCTGCTGTGCCCTCAGCCTGCCACATTCTGGAGTGTCCCGA\n",
    "AGGGCTTGCCCAGGATGTCATCAGCACCATTGGCCAGGCCTTCGAGTTGCGCTTCAAACAATACCTCAGG\n",
    "AACCCACCCAAACTGGTCACCCCTCATGACAGGTGAGCAGGTGGGGAGAGTGTGGAGCAGCTCTTGGATT\n",
    "AGGATTAGGAAAGGAGGGCTGCTCTTCCTACACTTGAGTTGAGAGGTTCAGGGAGGGAATAACCTGTGTG\n",
    "CCTCTGCTTCCCACTTCCTGCTATTCTTCTGTTTCCTGTGGCCCAGGAGGAAAGACTTTTGAGGCCCTGC\n",
    "TTGCCGCAGGACCAGCTTTTCTTGTTTGCTCATGGTGGAGTTGTAGTCCGGGGCCTCCAAGTTACATCTC\n",
    "TCTCTGGCTAGGTGGGATTTTCTACTCCCATCCCTCTTTCTTGCTCATTTCTCTGCTTGGCATTATGTGT\n",
    "TTTCCTGCCTCTAGAAGCCATCAAGAATTTGGGTTTGGGAGGATATCTCTTTTCCCTGAGATTACAACCT\n",
    "TTCCCACCTCAGGCCTTGAGGGATAAGCAGTGGAGCCAGTAGGGATTGAATATTTAGGAGAGGTTAAAAT\n",
    "ATTCTGGAGCTGGAGTGCCTCCCCCGGCCCCCCACCCCCCTACATGGTGCTTCCCTTCCTTGTTGGGAGA\n",
    "TAGCACAGAGATCAAGAGCAAGAGCATTACTTATTTGAGACAGTCTTGCTCTGTAGCCCAGGCTGGAGTT\n",
    "CAGTGGTACAATCACAGCTCACTGTAGCCTCCATATCCTGGGCTTGAGTGATCCTCCCACCTTAGCCTCC\n",
    "TGAGTAGCTGGGGCTACAGGTGCATGCCACCACACCTGGCTGAGTAGTAGTATTTTTATTCTGGTAGAGA\n",
    "TAAGGTCTCGCTATGTTGTCCTGGCTGGTCTTGAACTCCTAGGCTCAAACAACCCTCCCTCCTTGGCCTC\n",
    "TGAAAGTGTTGGGATTATAGGCGTGAGCCATTGCACCTGGCCAAGGGGATGAGCTTTAAAGTCAAACCTG\n",
    "AATTTAAATCTGAGCTCGGCCGCCTTTTAGTTGTGAGACCTTGGGCAGATCTTTTCACCTCTGACCCTCA\n",
    "TTTTCCTCATCTGTAAAAGTGGGGAAGGAAACAATACCTACCTTACGGGGTTGCCAGGAGAATGAAATGA\n",
    "TTATGTATATAAAACACTCAGCCTCAGGCTTAATCCATAGTAAATACCCAATAGATGGCTGCTGTTCTAT\n",
    "TCTGTCCTTCCCTCCTCCTGCCCACCACACATTTACAGCCACAGCTTCACTGTCCTGTCCCTCAGGATGG\n",
    "CTGGCTTTGATGGCTCAGCATGGGATGAGGAGGAGGAAGAGCCACCTGACCATCAGTACTATAATGACTT\n",
    "CCCGGGGAAGGAACCCCCCTTGGGGGGGGTGGTAGACATGAGGCTTCGGGAAGGAGCCGCTCCAGGGGCT\n",
    "GCTCGACCCACTGCACCCAATGCCCAGACCCCCAGCCACTTGGGAGCTACATTGGTAAGTTGCTTGGATG\n",
    "GAGGCGGGCTGGGGCTAGGGTGGGAGTGGTTGGTTAGGGCTTCTGGTCTCACCTTATTTTTATCCCTGCA\n",
    "GCCTGTAGGACAGCCTGTTGGGGGAGATCCAGAAGTCCGCAAACAGATGCCACCTCCACCACCCTGTCCA\n",
    "GGTATGAAGGGAGCCGAGATAGGCAGAGCTGGGGAGTGTTGGAGGCAGGGCTGGGATCACTGTAGAGCCT\n",
    "CCCTTTCCTCAGCAGGCAGAGAGCTTTTTGATGATCCCTCCTATGTCAACGTCCAGAACCTAGACAAGGC\n",
    "CCGGCAAGCAGTGGGTGGTGCTGGGCCCCCCAATCCTGCTATCAATGGCAGTGCACCCCGGGACCTGTTT\n",
    "GACATGAGTGAGTGCTCGTCTCTCTCCTCTCTGCATCCCCACCTTTCTACTTGGTTCTGGCCCTTCTCCT\n",
    "GCCTATCTGCTCTTTCCTTCCTGACTTCCCTTCCCTGAAGTCTGGGTCTTGGGTGTGTGTTGTGCCTGCT\n",
    "ACTTCCCCTCACCCTCCCTCTATACTTCCAGAGCCCTTCGAAGATGCTCTTCGCGTGCCTCCACCTCCCC\n",
    "AGTCGGTGTCCATGGCTGAGCAGCTCCGAGGGGAGCCCTGGTTCCATGGGAAGCTGAGCCGGCGGGAGGC\n",
    "TGAGGCACTGCTGCAGCTCAATGGGGACTTCCTGGTACGGGAGAGCACGACCACACCTGGCCAGTATGTG\n",
    "CTCACTGGCTTGCAGAGTGGGCAGCCTAAGCATTTGCTACTGGTGGACCCTGAGGGTGTGGTGAGTGTGG\n",
    "CAGAGGTGTGGGTGAGGGGTGGCAAAAAGGAAGGTACAGTACCCTACAGTGTATCCCTCTTCCTTCCCTC\n",
    "ATTGCTTTGTAGACATTCCCGCTGGGCAGCTGTCTATGGGCCTCTCTTGCTCTTTAAAGTGTCTGGCATC\n",
    "TTCCTCCAGAGGCTCATCCCTTAGCCTGAACTATGTTCTTTATAGTCCTGGCACAGGGCCAGGCGTGGTG\n",
    "GCTCACACCTGTAATCCCAGCGCTTTGGGAGGCTGAGGCAGGTGAATCACCTGAGGTCAGGAGTTCGAGA\n",
    "CCAGCCTGACCAATATGGTGAAACCCTGTCTCTACTAAAAATACAAAAATTAGCCGGGTGTGGTGGCAGG\n",
    "CACCTGTAGTCCCAGCTACTCAGGAGGCTGAGGCAGGAGAATTGCTTGAATCTGGGATGCAGAGGTTGCA\n",
    "ATGAGCTGAGATCACACCACTGCACTCCAGCCTGGGTGACAGAGTGAGACTCTGTCTGAAAAAACAAAAC\n",
    "AAAACAAATACAGTCCTGGCCCGTCCTCAAATCCCCAAGCCTTAAGCTTATCAGAGCTAGAAGGCAGTCA\n",
    "TCTCATCTGTGCCCCTCTTTTGACAAATAAGGAAACCACCCTGAAGAGGTTGTGATGCCCAATGTCAGAT\n",
    "GGTTAGAGGCTAGACACTGCAAGTTTCCTTTTGTAAGAAGATCCAAGGGCTTCTCCAGCTCCCCAGACTC\n",
    "AGGACTCAGACCACTGTGCTTTCTCTTATCTGCACTGCCTCACTGTTCTTCAAAACTTTCCTTGTTTAGC\n",
    "CAGGTGTGGTGGTTCATGCCTGTAATCCCAACATTTTGCGAAGCTGAGGTGGGAGGATTGCTTGAGGCCA\n",
    "GGAGTTCAAGACCAGCCTGGGCAACATAGCGAGACCTTGTCTCTAGTCAGGAATGGTGACACATGCCTGT\n",
    "AGTCCTACCTACCCAGGAGGCTGAGGCCTGAGGATCGCTTGAGCCCAGGAGTTCGAGGTTACAGTGAGCC\n",
    "ATGATTGCGCCACTGTACTCTAGCCTGGGTGACAGAGACCCTGTCTGGAAACAAAAAAACTTTTGTTTGT\n",
    "TTTCTACAGTAATCTCAGCTTCCTTTCTTTTTTTTTGAGATAGGGTTTTGCTTTGTCACTCAGGCTGCGG\n",
    "TGTGGTGGCACACCTGGCTAATTTTTGTATTTTTTTTTTGGTAGAGATGGGGTTTCGCCATGTTGCCCAA\n",
    "GCTGTTCTTGAACTCCCGGGCTCAAGTGATCCGCCTCCCTCGGCCTCCCAGAGTGCTGGGATTACAGGCA\n",
    "CGGACCACCATGCCCAGCCTCCACATCTTTTTTTGCACTGTGTATACTCTTCTGAGACATGCCAACTTCC\n",
    "TCCAGGTCAAGAAAGGGGTATATAGCTCTCAGCTTCACTCTTTCAGGGCTGATGTCGCCTTTGCCTTTTC\n",
    "TCACTTCACTGACCTGTCTATTCCTACAACTGTCTCTTTCTAGAGAAGCCTCAATGATCAGGATTGACAG\n",
    "GCCACACTCTCCCCCACCATTTTTTTCTCCTCCTTCAAGCCTCTTGTCTGTTTCACCCTCTTCCACCTTG\n",
    "GAGGCTGAGGTCTTATTTGACTCTTCACCTGAATTGACCTTCTTCCTTCCCACCCTCCCAGGTTCGGACT\n",
    "AAGGATCACCGCTTTGAAAGTGTCAGTCACCTTATCAGCTACCACATGGACAATCACTTGCCCATCATCT\n",
    "CTGCGGGCAGCGAACTGTGTCTACAGCAACCTGTGGAGCGGAAACTGTGATCTGCCCTAGCGCTCTCTTC\n",
    "CAGAAGATGCCCTCCAATCCTTTCCACCCTATTCCCTAACTCTCGGGACCTCGTTTGGGAGTGTTCTGTG\n",
    "GGCTTGGCCTTGTGTCAGAGCTGGGAGTAGCATGGACTCTGGGTTTCATATCCAGCTGAGTGAGAGGGTT\n",
    "TGAGTCAAAAGCCTGGGTGAGAATCCTGCCTCTCCCCAAACATTAATCACCAAAGTATTAATGTACAGAG\n",
    "TGGCCCCTCACCTGGGCCTTTCCTGTGCCAACCTGATGCCCCTTCCCCAAGAAGGTGAGTGCTTGTCATG\n",
    "GAAAATGTCCTGTGGTGACAGGCCCAGTGGAACAGTCACCCTTCTGGGCAAGGGGGAACAAATCACACCT\n",
    "CTGGGCTTCAGGGTATCCCAGACCCCTCTCAACACCCGCCCCCCCCATGTTTAAACTTTGTGCCTTTGAC\n",
    "CATCTCTTAGGTCTAATGATATTTTATGCAAACAGTTCTTGGACCCCTGAATTCAATGACAGGGATGCCA\n",
    "ACACCTTCTTGGCTTCTGGGACCTGTGTTCTTGCTGAGCACCCTCTCCGGTTTGGGTTGGGATAACAGAG\n",
    "GCAGGAGTGGCAGCTGTCCCCTCTCCCTGGGGATATGCAACCCTTAGAGATTGCCCCAGAGCCCCACTCC\n",
    "CGGCCAGGCGGGAGATGGACCCCTCCCTTGCTCAGTGCCTCCTGGCCGGGGCCCCTCACCCCAAGGGGTC\n",
    "TGTATATACATTTCATAAGGCCTGCCCTCCCATGTTGCATGCCTATGTACTCTACGCCAAAGTGCAGCCC\n",
    "TTCCTCCTGAAGCCTCTGCCCTGCCTCCCTTTCTGGGAGGGCGGGGTGGGGGTGACTGAATTTGGGCCTC\n",
    "TTGTACAGTTAACTCTCCCAGGTGGATTTTGTGGAGGTGAGAAAAGGGGCATTGAGACTATAAAGCAGTA\n",
    "GACAATCCCCACATACCATCTGTAGAGTTGGAACTGCATTCTTTTAAAGTTTTATATGCATATATTTTAG\n",
    "GGCTGTAGACTTACTTTCCTATTTTCTTTTCCATTGCTTATTCTTGAGCACAAAATGATAATCAATTATT\n",
    "ACATTTATACATCACCTTTTTGACTTTTCCAAGCCCTTTTACAGCTCTTGGCATTTTCCTCGCCTAGGCC\n",
    "TGTGAGGTAACTGGGATCGCACCTTTTATACCAGAGACCTGAGGCAGATGAAATTTATTTCCATCTAGGA\n",
    "CTAGAAAAACTTGGGTCTCTTACCGCGAGACTGAGAGGCAGAAGTCAGCCCGAATGCCTGTCAGTTTCAT\n",
    "GGAGGGGAAACGCAAAACCTGCAGTTCCTGAGTACCTTCTACAGGCCCGGCCCAGCCTAGGCCCGGGGTG\n",
    "GCCACACCACAGCAAGCCGGCCCCCCCTCTTTTGGCCTTGTGGATAAGGGAGAGTTGACCGTTTTCATCC\n",
    "TGGCCTCCTTTTGCTGTTTGGATGTTTCCACGGGTCTCACTTATACCAAAGGGAAAACTCTTCATTAAAG\n",
    "TCCGTATTTCTTCTA\"\"\"  \n",
    "seq = Seq(\"\".join(raw_sequence.split()).rstrip())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[Seq('LATTAFVRDLALPLANSGKWEA', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('RRDPEVEGQGLVRPEVSVIKFLQGGRARGESWSGNLSWQWRDPAPNRPAVRCGDSMRPW', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('VLTFFSLGDSLFHPGTPLPGPHFRSSFISPAPETPLPGSPTSQLQGAPPNPDLI...ILG', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('LLSKGLSPKRAKPCSSKNDSNLLKLRPSEATPPP', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('STSDCPESLPAQKSEVRVLRGPRPLRSFIFVGPGSSAEKPCRQSLKLHLLIFSC...CKE', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('D', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('RQPFSVFLSIFFFFFFFFKTEFRSYRPGWSAMVRSRLTATSAS', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('VQVILLPQPPE', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('LGLQAPTTTPG', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('FLYF', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('RRGFPMLARLASNS', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('PQVLHPYQPP', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('VLGLQALATAPVPVFLFLNLLQG', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('NYLKGLNVGFGTRGRERLALRLPASLPGFPFPGPTSSQK', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('L', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('AFLDISG', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('DHGFVGVWHVGVT', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('AKVFYYQKNGSLEWG', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('IIVLFLLFDFFFFKGQFSRYFSVSSPPPNQSSCLGIPPGPVCPAFPAGNIPEAF...LLE', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('ALRGDVRVKHSI', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('HTSSVNVVNQDNYLSIF', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('FSEVNLRVIPR', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('AGGNTYFFFFETEFCSCHLGWSAMARSRLTATSASWVRAILLPQPPELGLQVPATTPS', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('FLYF', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('RQGFTMLTRLVSNS', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('LQVIHPPQPPKVLVSQV', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('ATAPGRNTYLCFFFFGHFPALPVAFSGRAVAGSRGWQEWVWDCSLCPASTHLLL...PGL', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('FLLLLENCFFHLLQPGSFPGVKDAVLFGGA', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('GPVRGGM', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('IGVVWSGSGDLIASFSPLTVCCVNKPCGKGRQDSAWL', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('AIGLGVVDGIRREQAVTSMSSCSLLAW', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('LPDVSSRLNFIPLFSPALLFLLALLCIHWDHLLPIPTSLAPVHSCCSRGC', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('WPRLTVWGSGWGWPGLGPASFRA', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('RGWGNSSVQPGQEGKC', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('ALRSPPFSTPSCLCFLPHSPVLLPGFLPPLACVCPVLSDPDLLSPSFPSHSSIL...LSL', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('VSTVLTIGLSP', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('LPAPGEEPEALGPACSLLSQRRGSTLGGDGA', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('NCLGLSWGAEATCPSPSPGLL', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('LLGHRGPTRLRAWGYPLPGPLAQTGRGARLGSSPSFTSTMDLLPPKPKYNPLRN...LQF', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('CSLQLGRNLFFPAVPQQPRFSVLFPKPFSSLPRFCSPFPSSFAWAPFLLFTPPPPPWHA', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('HNAQHGRSS', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('VLE', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('MTS', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('SFLHSPPAFWSCHSIPANACRSPPNSLPSV', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('GPSSLPFLFLLIPCCLSPWLWDVATGSGQNFFLFTSLRDHMGKVLGPFVKGLGWGSR', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('PAVLFSLPLVHGLCGGPPVNACPGLQHPDSGHQVSGEGVDCGSPGISEAMGNRG...KLR', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('FRLLLKGGPFGPDPPYLSSLGSCGEGGRSSWQVELE', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('KRGVDGRLLLPLGREWDIW', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('LEGGGAIPIEALTQSARKWPLSVIKY', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('RPLIQSTMTKSLECPGSLAWLSPPPALWQPQLSGRQAGRRWV', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('VPLFPQGGHQSGV', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('GCAGC', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('GGDKEEKGTWGPWG', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('GCCWEMWRGSGEASVGHVGRGPGGKSVVCLCWEPSPSETLGPTLAL', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('PPAQLYPGEE', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('PEICWNANHSHRLHQQPQPHGRRLQTGW', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('RWADGPETLMGVLRGDLVRRPKSRKRA', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('EQETSREGGNEDMG', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('GTQRMRLGHQGSGP', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('GCLTHGIPFLLHPLP', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('FSDHRQPPHAIYLICIRRGSGELGNRAKAVGEEQGARDWEDID', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('RSGGARIVGISEKDWVLGVRSKVEL', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('NFLTFLPRTQPSMSPMLPKTL', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('IREVRHGGEHWVGLAV', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('GGE', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('DLWSSVPTLSRSQWGFLQRKMRAPVCASSQWAQLTDSLCCALSLPHSGVSRRAC...HPS', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('QVSRWGECGAALGLGLGKEGCSSYT', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('VERFREGITCVPLLPTSCYSSVSCGPGGKTFEALLAAGPAFLVCSWWSCSPGPP...LEG', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('AVEPVGIEYLGEVKIFWSWSASPGPPPPYMVLPFLVGR', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('HRDQEQEHYLFETVLLCSPGWSSVVQSQLTVASISWA', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('VILPP', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('PPE', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('LGLQVHATTPG', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('VVVFLFW', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('R', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('GLAMLSWLVLNS', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('AQTTLPPWPLKVLGL', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('A', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('AIAPGQGDEL', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('SQT', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('I', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('I', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('ARPPFSCETLGRSFHL', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('PSFSSSVKVGKETIPTLRGCQENEMIMYIKHSASGLIHSKYPIDGCCSILSFPP...GGW', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('T', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('GFGKEPLQGLLDPLHPMPRPPATWELHW', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('VAWMEAGWG', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('GGSGWLGLLVSPYFYPCSL', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('DSLLGEIQKSANRCHLHHPVQV', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('REPR', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('AELGSVGGRAGITVEPPFPQQAESFLMIPPMSTSRT', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('TRPGKQWVVLGPPILLSMAVHPGTCLT', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('VSARLSPLCIPTFLLGSGPSPAYLLFPS', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('LPFPEVWVLGVCCACYFPSPSLYTSRALRRCSSRASTSPVGVHG', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('AAPRGALVPWEAEPAGG', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('GTAAAQWGLPGTGEHDHTWPVCAHWLAEWAA', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('AFATGGP', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('GCGECGRGVGEGWQKGRYSTLQCIPLPSLIAL', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('TFPLGSCLWASLAL', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('SVWHLPPEAHPLA', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('TMFFIVLAQGQAWWLTPVIPALWEAEAGESPEVRSSRPA', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('PIW', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('NPVSTKNTKISRVWWQAPVVPATQEAEAGELLESGMQRLQ', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('AEITPLHSSLGDRVRLCLKKQNKTNTVLARPQIPKP', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('AYQS', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('KAVISSVPLF', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('QIRKPP', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('RGCDAQCQMVRG', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('TLQVSFCKKIQGLLQLPRLRTQTTVLSLICTASLFFKTFLV', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('PGVVVHACNPNILRS', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('GGRIA', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('GQEFKTSLGNIARPCL', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('SGMVTHACSPTYPGG', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('GLRIA', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('AQEFEVTVSHDCATVL', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('PG', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('QRPCLETKKLLFVFYSNLSFLSFFLR', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('GFALSLRLRCGGTPG', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('FLYFFFGRDGVSPCCPSCS', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('TPGLK', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('SASLGLPECWDYRHGPPCPASTSFFALCILF', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('DMPTSSRSRKGYIALSFTLSGLMSPLPFLTSLTCLFLQLSLSREASMIRIDRPH...GSD', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('GSPL', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('KCQSPYQLPHGQSLAHHLCGQRTVSTATCGAETVICPSALFQKMPSNPFHPIP', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('LSGPRLGVFCGLGLVSELGVAWTLGFISS', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('VRGFESKAWVRILPLPKH', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('SPKY', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('CTEWPLTWAFPVPT', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('CPFPKKVSACHGKCPVVTGPVEQSPFWARGNKSHLWASGYPRPLSTPAPPMFKL...GLG', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('QRQEWQLSPLPGDMQPLEIAPEPHSRPGGRWTPPLLSASWPGPLTPRGLYIHFI...PSS', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('SLCPASLSGRAGWG', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('LNLGLLYS', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('LSQVDFVEVRKGALRL', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('SSRQSPHTICRVGTAFF', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('SFICIYFRAVDLLSYFLFHCLFLSTK', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('SIITFIHHLFDFSKPFYSSWHFPRLGL', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('GNWDRTFYTRDLRQMKFISI', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('D', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('KNLGLLPRD', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('EAEVSPNACQFHGGETQNLQFLSTFYRPGPA', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('ARGGHTTASRPPLFWPCG', HasStopCodon(ExtendedIUPACProtein(), '*')), Seq('GRVDRFHPGLLLLFGCFHGSHLYQRENSSLKSVFLL', HasStopCodon(ExtendedIUPACProtein(), '*'))]\n"
     ]
    }
   ],
   "source": [
    "messenger_rna = seq.transcribe()\n",
    "transcripts = messenger_rna.translate().split(\"*\")\n",
    "\n",
    "print([transcript for transcript in transcripts])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Standard\n",
      "Alternative Flatworm Mitochondrial\n",
      "SGC1\n",
      "Plant Plastid\n",
      "SGC0\n",
      "Mold Mitochondrial\n",
      "Hexamita Nuclear\n",
      "SGC2\n",
      "Flatworm Mitochondrial\n",
      "SGC5\n",
      "Ciliate Nuclear\n",
      "Bacterial\n",
      "Scenedesmus obliquus Mitochondrial\n",
      "Coelenterate Mitochondrial\n",
      "Chlorophycean Mitochondrial\n",
      "Spiroplasma\n",
      "Trematode Mitochondrial\n",
      "Protozoan Mitochondrial\n",
      "Echinoderm Mitochondrial\n",
      "Ascidian Mitochondrial\n",
      "Alternative Yeast Nuclear\n",
      "SGC4\n",
      "SGC8\n",
      "Euplotid Nuclear\n",
      "Dasycladacean Nuclear\n",
      "Blepharisma Macronuclear\n",
      "Mycoplasma\n",
      "Vertebrate Mitochondrial\n",
      "Thraustochytrium Mitochondrial\n",
      "SGC3\n",
      "SGC9\n",
      "Invertebrate Mitochondrial\n",
      "Yeast Mitochondrial\n",
      "Pterobranchia Mitochondrial\n",
      "Table 1 Standard, SGC0\n",
      "\n",
      "  |  T      |  C      |  A      |  G      |\n",
      "--+---------+---------+---------+---------+--\n",
      "T | TTT F   | TCT S   | TAT Y   | TGT C   | T\n",
      "T | TTC F   | TCC S   | TAC Y   | TGC C   | C\n",
      "T | TTA L   | TCA S   | TAA Stop| TGA Stop| A\n",
      "T | TTG L(s)| TCG S   | TAG Stop| TGG W   | G\n",
      "--+---------+---------+---------+---------+--\n",
      "C | CTT L   | CCT P   | CAT H   | CGT R   | T\n",
      "C | CTC L   | CCC P   | CAC H   | CGC R   | C\n",
      "C | CTA L   | CCA P   | CAA Q   | CGA R   | A\n",
      "C | CTG L(s)| CCG P   | CAG Q   | CGG R   | G\n",
      "--+---------+---------+---------+---------+--\n",
      "A | ATT I   | ACT T   | AAT N   | AGT S   | T\n",
      "A | ATC I   | ACC T   | AAC N   | AGC S   | C\n",
      "A | ATA I   | ACA T   | AAA K   | AGA R   | A\n",
      "A | ATG M(s)| ACG T   | AAG K   | AGG R   | G\n",
      "--+---------+---------+---------+---------+--\n",
      "G | GTT V   | GCT A   | GAT D   | GGT G   | T\n",
      "G | GTC V   | GCC A   | GAC D   | GGC G   | C\n",
      "G | GTA V   | GCA A   | GAA E   | GGA G   | A\n",
      "G | GTG V   | GCG A   | GAG E   | GGG G   | G\n",
      "--+---------+---------+---------+---------+--\n",
      "Table 2 Vertebrate Mitochondrial, SGC1\n",
      "\n",
      "  |  T      |  C      |  A      |  G      |\n",
      "--+---------+---------+---------+---------+--\n",
      "T | TTT F   | TCT S   | TAT Y   | TGT C   | T\n",
      "T | TTC F   | TCC S   | TAC Y   | TGC C   | C\n",
      "T | TTA L   | TCA S   | TAA Stop| TGA W   | A\n",
      "T | TTG L   | TCG S   | TAG Stop| TGG W   | G\n",
      "--+---------+---------+---------+---------+--\n",
      "C | CTT L   | CCT P   | CAT H   | CGT R   | T\n",
      "C | CTC L   | CCC P   | CAC H   | CGC R   | C\n",
      "C | CTA L   | CCA P   | CAA Q   | CGA R   | A\n",
      "C | CTG L   | CCG P   | CAG Q   | CGG R   | G\n",
      "--+---------+---------+---------+---------+--\n",
      "A | ATT I(s)| ACT T   | AAT N   | AGT S   | T\n",
      "A | ATC I(s)| ACC T   | AAC N   | AGC S   | C\n",
      "A | ATA M(s)| ACA T   | AAA K   | AGA Stop| A\n",
      "A | ATG M(s)| ACG T   | AAG K   | AGG Stop| G\n",
      "--+---------+---------+---------+---------+--\n",
      "G | GTT V   | GCT A   | GAT D   | GGT G   | T\n",
      "G | GTC V   | GCC A   | GAC D   | GGC G   | C\n",
      "G | GTA V   | GCA A   | GAA E   | GGA G   | A\n",
      "G | GTG V(s)| GCG A   | GAG E   | GGG G   | G\n",
      "--+---------+---------+---------+---------+--\n",
      "Table 11 Bacterial, Plant Plastid\n",
      "\n",
      "  |  T      |  C      |  A      |  G      |\n",
      "--+---------+---------+---------+---------+--\n",
      "T | TTT F   | TCT S   | TAT Y   | TGT C   | T\n",
      "T | TTC F   | TCC S   | TAC Y   | TGC C   | C\n",
      "T | TTA L   | TCA S   | TAA Stop| TGA Stop| A\n",
      "T | TTG L(s)| TCG S   | TAG Stop| TGG W   | G\n",
      "--+---------+---------+---------+---------+--\n",
      "C | CTT L   | CCT P   | CAT H   | CGT R   | T\n",
      "C | CTC L   | CCC P   | CAC H   | CGC R   | C\n",
      "C | CTA L   | CCA P   | CAA Q   | CGA R   | A\n",
      "C | CTG L(s)| CCG P   | CAG Q   | CGG R   | G\n",
      "--+---------+---------+---------+---------+--\n",
      "A | ATT I(s)| ACT T   | AAT N   | AGT S   | T\n",
      "A | ATC I(s)| ACC T   | AAC N   | AGC S   | C\n",
      "A | ATA I(s)| ACA T   | AAA K   | AGA R   | A\n",
      "A | ATG M(s)| ACG T   | AAG K   | AGG R   | G\n",
      "--+---------+---------+---------+---------+--\n",
      "G | GTT V   | GCT A   | GAT D   | GGT G   | T\n",
      "G | GTC V   | GCC A   | GAC D   | GGC G   | C\n",
      "G | GTA V   | GCA A   | GAA E   | GGA G   | A\n",
      "G | GTG V(s)| GCG A   | GAG E   | GGG G   | G\n",
      "--+---------+---------+---------+---------+--\n"
     ]
    }
   ],
   "source": [
    "from Bio.Data import CodonTable\n",
    "for i in CodonTable.unambiguous_dna_by_name:\n",
    "    print(i)\n",
    "print(CodonTable.unambiguous_dna_by_name[\"Standard\"])\n",
    "print(CodonTable.unambiguous_dna_by_name[\"Vertebrate Mitochondrial\"])\n",
    "print(CodonTable.unambiguous_dna_by_name[\"Bacterial\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['__add__', '__class__', '__contains__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getitem__', '__gt__', '__hash__', '__init__', '__le__', '__len__', '__lt__', '__module__', '__ne__', '__new__', '__radd__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_data', '_get_seq_str_and_check_alphabet', 'alphabet', 'back_transcribe', 'complement', 'count', 'endswith', 'find', 'lower', 'lstrip', 'reverse_complement', 'rfind', 'rsplit', 'rstrip', 'split', 'startswith', 'strip', 'tomutable', 'tostring', 'transcribe', 'translate', 'ungap', 'upper']\n"
     ]
    }
   ],
   "source": [
    "print(dir(seq))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A: 2418\n",
      "C: 3377\n",
      "G: 3189\n",
      "T: 3211\n",
      "All: 12196\n",
      "All: 12195\n"
     ]
    }
   ],
   "source": [
    "print(\"A:\", seq.count(\"A\"))\n",
    "print(\"C:\", seq.count(\"C\"))\n",
    "print(\"G:\", seq.count(\"G\"))\n",
    "print(\"T:\", seq.count(\"T\"))\n",
    "print(\"All:\", seq.count(\"\"))\n",
    "print(\"All:\", seq.count(\"A\") + seq.count(\"C\") + seq.count(\"G\") + seq.count(\"T\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12195\n"
     ]
    }
   ],
   "source": [
    "print(len(seq))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Norovirus\n",
      "Seq('APPTLWSRVTKFGSGWGFWVSPTVFITTTHVVPTGVKEFFGEPLSSIAIHQAGE...ALE', SingleLetterAlphabet())\n",
      "181\n",
      "\n",
      "tr|Q9YQ30|Q9YQ30_9PLVG\n",
      "Seq('PQITLWQRPLVTIKIGGQLKEALLDTGADDTVLEEMNLPGRWKPKMIGGIGGFI...LNF', SingleLetterAlphabet())\n",
      "99\n",
      "\n",
      "tr|F8RDC3|F8RDC3_HHV1\n",
      "Seq('MAADAPGDRMEEPLPDRAVPIYVAGFLALYDSGDSGELALDPDTVRAALPPDNP...GAR', SingleLetterAlphabet())\n",
      "635\n",
      "\n",
      "2A\n",
      "Seq('GPSDMYVHVGNLIYRNLHLFNSEMHESILVSYSSDLIIYRTNTVGDDYIPSCDC...EEQ', SingleLetterAlphabet())\n",
      "142\n",
      "\n",
      "3C\n",
      "Seq('GPEEEFGMSLIKHNSCVITTENGKFTGLGVYDRFVVVPTHADPGKEIQVDGITT...TDV', SingleLetterAlphabet())\n",
      "182\n",
      "\n",
      "tev\n",
      "Seq('MALIFGTVNANILKEVFGGARMACVTSAHMAGANGSILKKAEETSRAIMHKPVI...THY', SingleLetterAlphabet())\n",
      "304\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "sequences = []\n",
    "for seq_record in SeqIO.parse(\"proteases.fasta\", \"fasta\"):\n",
    "    sequences.append(seq_record)\n",
    "    print(seq_record.id)\n",
    "    print(repr(seq_record.seq))\n",
    "    print(len(seq_record))\n",
    "    print()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Importing #"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n",
      "<!DOCTYPE eInfoResult PUBLIC \"-//NLM//DTD einfo 20130322//EN\" \"http://eutils.ncbi.nlm.nih.gov/eutils/dtd/20130322/einfo.dtd\">\n",
      "<eInfoResult>\n",
      "<DbList>\n",
      "\n",
      "\t<DbName>pubmed</DbName>\n",
      "\t<DbName>protein</DbName>\n",
      "\t<DbName>nuccore</DbName>\n",
      "\t<DbName>nucleotide</DbName>\n",
      "\t<DbName>nucgss</DbName>\n",
      "\t<DbName>nucest</DbName>\n",
      "\t<DbName>structure</DbName>\n",
      "\t<DbName>genome</DbName>\n",
      "\t<DbName>annotinfo</DbName>\n",
      "\t<DbName>assembly</DbName>\n",
      "\t<DbName>bioproject</DbName>\n",
      "\t<DbName>biosample</DbName>\n",
      "\t<DbName>blastdbinfo</DbName>\n",
      "\t<DbName>books</DbName>\n",
      "\t<DbName>cdd</DbName>\n",
      "\t<DbName>clinvar</DbName>\n",
      "\t<DbName>clone</DbName>\n",
      "\t<DbName>gap</DbName>\n",
      "\t<DbName>gapplus</DbName>\n",
      "\t<DbName>grasp</DbName>\n",
      "\t<DbName>dbvar</DbName>\n",
      "\t<DbName>epigenomics</DbName>\n",
      "\t<DbName>gene</DbName>\n",
      "\t<DbName>gds</DbName>\n",
      "\t<DbName>geoprofiles</DbName>\n",
      "\t<DbName>homologene</DbName>\n",
      "\t<DbName>medgen</DbName>\n",
      "\t<DbName>mesh</DbName>\n",
      "\t<DbName>ncbisearch</DbName>\n",
      "\t<DbName>nlmcatalog</DbName>\n",
      "\t<DbName>omim</DbName>\n",
      "\t<DbName>orgtrack</DbName>\n",
      "\t<DbName>pmc</DbName>\n",
      "\t<DbName>popset</DbName>\n",
      "\t<DbName>probe</DbName>\n",
      "\t<DbName>proteinclusters</DbName>\n",
      "\t<DbName>pcassay</DbName>\n",
      "\t<DbName>biosystems</DbName>\n",
      "\t<DbName>pccompound</DbName>\n",
      "\t<DbName>pcsubstance</DbName>\n",
      "\t<DbName>pubmedhealth</DbName>\n",
      "\t<DbName>seqannot</DbName>\n",
      "\t<DbName>snp</DbName>\n",
      "\t<DbName>sra</DbName>\n",
      "\t<DbName>taxonomy</DbName>\n",
      "\t<DbName>unigene</DbName>\n",
      "\t<DbName>gencoll</DbName>\n",
      "\t<DbName>gtr</DbName>\n",
      "</DbList>\n",
      "\n",
      "</eInfoResult>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.email = \"xtals8hua@gmail.com\"\n",
    "handle = Entrez.einfo()\n",
    "result = handle.read()\n",
    "\n",
    "print(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pubmed\n",
      "protein\n",
      "nuccore\n",
      "nucleotide\n",
      "nucgss\n",
      "nucest\n",
      "structure\n",
      "genome\n",
      "annotinfo\n",
      "assembly\n",
      "bioproject\n",
      "biosample\n",
      "blastdbinfo\n",
      "books\n",
      "cdd\n",
      "clinvar\n",
      "clone\n",
      "gap\n",
      "gapplus\n",
      "grasp\n",
      "dbvar\n",
      "epigenomics\n",
      "gene\n",
      "gds\n",
      "geoprofiles\n",
      "homologene\n",
      "medgen\n",
      "mesh\n",
      "ncbisearch\n",
      "nlmcatalog\n",
      "omim\n",
      "orgtrack\n",
      "pmc\n",
      "popset\n",
      "probe\n",
      "proteinclusters\n",
      "pcassay\n",
      "biosystems\n",
      "pccompound\n",
      "pcsubstance\n",
      "pubmedhealth\n",
      "seqannot\n",
      "snp\n",
      "sra\n",
      "taxonomy\n",
      "unigene\n",
      "gencoll\n",
      "gtr\n"
     ]
    }
   ],
   "source": [
    "import xml.etree.ElementTree as ET\n",
    "\n",
    "root = ET.fromstring(result)\n",
    "\n",
    "# for child in root:\n",
    "#    print(child.tag, child.attrib)\n",
    "\n",
    "for i in range(len(root[0])):\n",
    "    print(root[0][i].text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['24929426', '24497503', '24267035', '24194598', '23842806', '23157543', '22909249', '22399473', '21666252', '21210977', '20015970', '19811691', '19773334', '19304878', '18606172', '21585724', '16403221', '16377612', '14871861', '14630660']\n"
     ]
    }
   ],
   "source": [
    "handle = Entrez.esearch(db=\"pubmed\", term=\"biopython\")\n",
    "record = Entrez.read(handle)\n",
    "print(record[\"IdList\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__iter__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_handle', 'close', 'read', 'readline', 'seek', 'tell', 'url']\n"
     ]
    }
   ],
   "source": [
    "\n",
    "handle = Entrez.efetch(db=\"pubmed\", id=\"24929426\")\n",
    "print(  dir(handle) )\n",
    "# record = Entrez.read(handle)\n",
    "# print(record)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Sequences again #"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "id:  Norovirus\n",
      "seq:  APPTLWSRVTKFGSGWGFWVSPTVFITTTHVVPTGVKEFFGEPLSSIAIHQAGEFTQFRFSKKMRPDLTGMVLEEGCPEGTVCSVLIKRDSGELLPLAVRMGAIASMRIQGRLVHGQSGMLLTGANAKGMDLGTIPGDCGAPYVHKRGNDWVVCGVHAAATKSGNTVVCAVQAGEGETALE\n",
      "name:  Norovirus\n",
      "description:   Norovirus sp|Q83883|1101-1281\n",
      "letter_annotations:  {}\n",
      "annotations:  {}\n",
      "features:  []\n",
      "dbxrefs:  []\n",
      "\n",
      "id:  tr|Q9YQ30|Q9YQ30_9PLVG\n",
      "seq:  PQITLWQRPLVTIKIGGQLKEALLDTGADDTVLEEMNLPGRWKPKMIGGIGGFIKVRQYDQILIEICGKKAIGTVLVGPTPVNIIGRNLLTQIGCTLNF\n",
      "name:  tr|Q9YQ30|Q9YQ30_9PLVG\n",
      "description:  tr|Q9YQ30|Q9YQ30_9PLVG HIV-1 protease (Fragment) OS=Human immunodeficiency virus GN=HIV-1 protease PE=2 SV=1\n",
      "letter_annotations:  {}\n",
      "annotations:  {}\n",
      "features:  []\n",
      "dbxrefs:  []\n",
      "\n",
      "id:  tr|F8RDC3|F8RDC3_HHV1\n",
      "seq:  MAADAPGDRMEEPLPDRAVPIYVAGFLALYDSGDSGELALDPDTVRAALPPDNPLPINVDHRAGCEVGRVLAVVDDPRGPFFVGLIACVQLERVLETAASAAIFERRGPPLSREERLLYLITNYLPSVSLATKRLGGEAHPDRTLFAHVALCAIGRRLGTIVTYDTGLDAAIAPFRHLSPASREGARRLAAEAELALSGRTWAPGVEALTHTLLSTAVNNMMLRDRWSLVAERRRQAGIAGHTYLQASEKFKMWGAEPVSAPARGYKNGAPESTDIPPGSIAAAPQGDRCPIVRQRGVASPPVLPPMNPVPTSGTPAPAPPGDGSYLWIPASHYNQLVAGHAAPQPQPHSAFGFPAAAGAVAYGPHGAGLSQHYPPHVAHQYPGVLFSGPSPLEAQIAALVGAIAADRQAGGQPAAGDPGVRGSGKRRRYEAGPSESYCDQDEPDADYPYYPGEARGGPRGVDSRRAARQSPGTNETITALMGAVTSLQQELAHMRARTSAPYGMYTPVAHYRPQVGEPEPTTTHPALCPPEAVYRPPPHSAPYGPPQGPASHAPTPPYAPAACPPGPPPPPCPSTQTRAPLPTEPAFPPAATGSQPEASNAEAGALVNASSAAHVDVDTARAADLFVSQMMGAR\n",
      "name:  tr|F8RDC3|F8RDC3_HHV1\n",
      "description:  tr|F8RDC3|F8RDC3_HHV1 Capsid maturation protease OS=Human herpesvirus 1 GN=UL26 PE=4 SV=1\n",
      "letter_annotations:  {}\n",
      "annotations:  {}\n",
      "features:  []\n",
      "dbxrefs:  []\n",
      "\n",
      "id:  2A\n",
      "seq:  GPSDMYVHVGNLIYRNLHLFNSEMHESILVSYSSDLIIYRTNTVGDDYIPSCDCTQATYYCKHKNRYFPITVTSHDWYEIQESEYYPKHIQYNLLIGEGPCEPGDCGGKLLCKHGVIGIVTAGGDNHVAFIDLRHFHCAEEQ\n",
      "name:  2A\n",
      "description:   2A rhinovirus2 sp|P04936|851-992\n",
      "letter_annotations:  {}\n",
      "annotations:  {}\n",
      "features:  []\n",
      "dbxrefs:  []\n",
      "\n",
      "id:  3C\n",
      "seq:  GPEEEFGMSLIKHNSCVITTENGKFTGLGVYDRFVVVPTHADPGKEIQVDGITTKVIDSYDLYNKNGIKLEITVLKLDRNEKFRDIRRYIPNNEDDYPNCNLALLANQPEPTIINVGDVVSYGNILLSGNQTARMLKYSYPTKSGYCGGVLYKIGQVLGIHVGGNGRDGFSAMLLRSYFTDV\n",
      "name:  3C\n",
      "description:   3C rhinovirus2 sp|P04936|1508-1689\n",
      "letter_annotations:  {}\n",
      "annotations:  {}\n",
      "features:  []\n",
      "dbxrefs:  []\n",
      "\n",
      "id:  tev\n",
      "seq:  MALIFGTVNANILKEVFGGARMACVTSAHMAGANGSILKKAEETSRAIMHKPVIFGEDYITEADLPYTPLHLEVDAEMERMYYLGRRALTHGKRRKVSVNNKRNRRRKVAKTYVGRDSIVEKIVVPHTERKVDTTAAVEDICNEATTQLVHNSMPKRKKQKNFLPATSLSNVYAQTWSIVRKRHMQVEIISKKSVRARVKRFEGSVQLFASVRHMYGERKRVDLRIDNWQQETLLDLAKRFKNERVDQSKLTFGSSGLVLRQGSYGPAHWYRHGMFIVRGRSDGMLVDARAKVTFAVCHSMTHY\n",
      "name:  tev\n",
      "description:   tev sp|P04517|1-304\n",
      "letter_annotations:  {}\n",
      "annotations:  {}\n",
      "features:  []\n",
      "dbxrefs:  []\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for sequence in sequences:\n",
    "    print(\"id: \", sequence.id)\n",
    "    print(\"seq: \", sequence.seq)\n",
    "    print(\"name: \", sequence.name)\n",
    "    print(\"description: \", sequence.description)\n",
    "    print(\"letter_annotations: \", sequence.letter_annotations)\n",
    "    print(\"annotations: \", sequence.annotations)\n",
    "    print(\"features: \", sequence.features)\n",
    "    print(\"dbxrefs: \", sequence.dbxrefs)\n",
    "    print()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Alignment #"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SingleLetterAlphabet() alignment with 6 rows and 635 columns\n",
      "--------------------------------------------...--- tr|Q9YQ30|Q9YQ30_9PLVG\n",
      "MAADAPGDRMEEPLPDRAVPIYVAGFLALYDSGDSGELALDPDT...GAR tr|F8RDC3|F8RDC3_HHV1\n",
      "--------------------------------------------...--- Norovirus\n",
      "--------------------------------------------...--- tev\n",
      "--------------------------------------------...--- 3C\n",
      "--------------------------------------------...--- 2A\n"
     ]
    }
   ],
   "source": [
    "from Bio import AlignIO\n",
    "# help(AlignIO)\n",
    "alignments = AlignIO.read(\"proteases.aln\", \"clustal\")\n",
    "print(alignments)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: tr|Q9YQ30|Q9YQ30_9PLVG\n",
      "Name: <unknown name>\n",
      "Description: tr|Q9YQ30|Q9YQ30_9PLVG\n",
      "Number of features: 0\n",
      "Seq('------------------------------------------------------...---', SingleLetterAlphabet())\n",
      "\n",
      "ID: tr|F8RDC3|F8RDC3_HHV1\n",
      "Name: <unknown name>\n",
      "Description: tr|F8RDC3|F8RDC3_HHV1\n",
      "Number of features: 0\n",
      "Seq('MAADAPGDRMEEPLPDRAVPIYVAGFLALYDSGDSGELALDPDTVRAALPPDNP...GAR', SingleLetterAlphabet())\n",
      "\n",
      "ID: Norovirus\n",
      "Name: <unknown name>\n",
      "Description: Norovirus\n",
      "Number of features: 0\n",
      "Seq('------------------------------------------------------...---', SingleLetterAlphabet())\n",
      "\n",
      "ID: tev\n",
      "Name: <unknown name>\n",
      "Description: tev\n",
      "Number of features: 0\n",
      "Seq('------------------------------------------------------...---', SingleLetterAlphabet())\n",
      "\n",
      "ID: 3C\n",
      "Name: <unknown name>\n",
      "Description: 3C\n",
      "Number of features: 0\n",
      "Seq('------------------------------------------------------...---', SingleLetterAlphabet())\n",
      "\n",
      "ID: 2A\n",
      "Name: <unknown name>\n",
      "Description: 2A\n",
      "Number of features: 0\n",
      "Seq('------------------------------------------------------...---', SingleLetterAlphabet())\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for alignment in alignments:\n",
    "    print(alignment)\n",
    "    print()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# swissprot #"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', 'accessions', 'annotation_update', 'comments', 'created', 'cross_references', 'data_class', 'description', 'entry_name', 'features', 'gene_name', 'host_organism', 'host_taxonomy_id', 'keywords', 'molecule_type', 'organelle', 'organism', 'organism_classification', 'references', 'seqinfo', 'sequence', 'sequence_length', 'sequence_update', 'taxonomy_id']\n",
      "accessions  ['P29353', 'B5BU19', 'D3DV78', 'O15290', 'Q5T180', 'Q5T183', 'Q5T184', 'Q5T185', 'Q5T186', 'Q8N4K5', 'Q96CL1']\n",
      "annotation_update  ('11-NOV-2015', 0)\n",
      "comments  ['FUNCTION: Signaling adapter that couples activated growth factor receptors to signaling pathways. Participates in a signaling cascade initiated by activated KIT and KITLG/SCF. Isoform p46Shc and isoform p52Shc, once phosphorylated, couple activated receptor tyrosine kinases to Ras via the recruitment of the GRB2/SOS complex and are implicated in the cytoplasmic propagation of mitogenic signals. Isoform p46Shc and isoform p52Shc may thus function as initiators of the Ras signaling cascade in various non-neuronal systems. Isoform p66Shc does not mediate Ras activation, but is involved in signal transduction pathways that regulate the cellular response to oxidative stress and life span. Isoform p66Shc acts as a downstream target of the tumor suppressor p53 and is indispensable for the ability of stress-activated p53 to induce elevation of intracellular oxidants, cytochrome c release and apoptosis. The expression of isoform p66Shc has been correlated with life span (By similarity). Participates in signaling downstream of the angiopoietin receptor TEK/TIE2, and plays a role in the regulation of endothelial cell migration and sprouting angiogenesis. {ECO:0000250, ECO:0000269|PubMed:14665640}.', 'SUBUNIT: Interacts with CPNE3; this interaction may mediate the binding of CPNE3 with ERBB2 (PubMed:20010870). Interacts with the NPXY motif of tyrosine-phosphorylated IGF1R and INSR in vitro via the PID domain. Once activated, binds to GRB2. Interacts with tyrosine-phosphorylated CD3T and DDR2. Interacts with the N- terminal region of APS. Interacts with phosphorylated LRP1 and IRS4. Interacts with INPP5D/SHIP1 and INPPL1/SHIP2. Interacts with TRIM31. Interacts with PTPN6/SHP (tyrosine phosphorylated). Identified in a complex containing FGFR4, NCAM1, CDH2, PLCG1, FRS2, SRC, SHC1, GAP43 and CTT (By similarity). Interacts with ALK, GAB2, GRB7 and KIT. Interacts with FLT4 (tyrosine- phosphorylated). Interacts with EPHB1 and GRB2; activates the MAPK/ERK cascade to regulate cell migration. Interacts with PDGFRB (tyrosine-phosphorylated). Interacts with ERBB4. Interacts with TEK/TIE2 (tyrosine-phosphorylated). Interacts with the Trk receptors NTRK1, NTRK2 and NTRK3; in a phosphotyrosine-dependent manner. Interacts with PTK2/FAK1. Interacts with herpes simplex virus 1 UL46. {ECO:0000250, ECO:0000269|PubMed:10194451, ECO:0000269|PubMed:10867024, ECO:0000269|PubMed:11349134, ECO:0000269|PubMed:11980671, ECO:0000269|PubMed:12925710, ECO:0000269|PubMed:14665640, ECO:0000269|PubMed:15102829, ECO:0000269|PubMed:15272003, ECO:0000269|PubMed:15316024, ECO:0000269|PubMed:15488758, ECO:0000269|PubMed:17274988, ECO:0000269|PubMed:19172738, ECO:0000269|PubMed:20010870, ECO:0000269|PubMed:23946459, ECO:0000269|PubMed:7537849, ECO:0000269|PubMed:7541045, ECO:0000269|PubMed:7544002, ECO:0000269|PubMed:7559478, ECO:0000269|PubMed:8155326, ECO:0000269|PubMed:8195171, ECO:0000269|PubMed:8874179, ECO:0000269|PubMed:8940081, ECO:0000269|PubMed:9038210, ECO:0000269|PubMed:9233773, ECO:0000269|PubMed:9488479, ECO:0000269|PubMed:9660833}.', 'INTERACTION: P05067:APP; NbExp=5; IntAct=EBI-78835, EBI-77613; P10275:AR; NbExp=14; IntAct=EBI-78835, EBI-608057; P46108:CRK; NbExp=3; IntAct=EBI-78835, EBI-886; P00533:EGFR; NbExp=26; IntAct=EBI-78835, EBI-297353; P04626:ERBB2; NbExp=9; IntAct=EBI-78835, EBI-641062; P21860:ERBB3; NbExp=5; IntAct=EBI-78835, EBI-720706; Q15303:ERBB4; NbExp=2; IntAct=EBI-78835, EBI-80371; P03372-4:ESR1; NbExp=2; IntAct=EBI-78835, EBI-4309277; Q13480:GAB1; NbExp=9; IntAct=EBI-78835, EBI-517684; P62993:GRB2; NbExp=23; IntAct=EBI-78835, EBI-401755; P08069:IGF1R; NbExp=2; IntAct=EBI-1000553, EBI-475981; P06213:INSR; NbExp=2; IntAct=EBI-78835, EBI-475899; Q62120:Jak2 (xeno); NbExp=2; IntAct=EBI-78835, EBI-646604; P10721:KIT; NbExp=8; IntAct=EBI-78835, EBI-1379503; P08581:MET; NbExp=5; IntAct=EBI-78835, EBI-1039152; P03495:NS (xeno); NbExp=2; IntAct=EBI-78835, EBI-2548993; P27986:PIK3R1; NbExp=3; IntAct=EBI-78835, EBI-79464; Q05209:PTPN12; NbExp=4; IntAct=EBI-78835, EBI-2266035; Q07889:SOS1; NbExp=2; IntAct=EBI-78835, EBI-297487;', 'SUBCELLULAR LOCATION: Cytoplasm.', 'SUBCELLULAR LOCATION: Isoform p46Shc: Mitochondrion matrix {ECO:0000269|PubMed:14573619}. Note=Localized to the mitochondria matrix. Targeting of isoform p46Shc to mitochondria is mediated by its first 32 amino acids, which behave as a bona fide mitochondrial targeting sequence. Isoform p52Shc and isoform p66Shc, that contain the same sequence but more internally located, display a different subcellular localization.', \"SUBCELLULAR LOCATION: Isoform p66Shc: Mitochondrion {ECO:0000250}. Note=In case of oxidative conditions, phosphorylation at 'Ser-36' of isoform p66Shc, leads to mitochondrial accumulation. {ECO:0000250}.\", 'ALTERNATIVE PRODUCTS: Event=Alternative promoter usage, Alternative splicing; Named isoforms=6; Name=p66Shc;   IsoId=P29353-1; Sequence=Displayed;   Note=Regulated by epigenetic modifications of its promoter   region.; Name=p52Shc;   IsoId=P29353-2; Sequence=VSP_016108;   Note=Contains a N-acetylmethionine at position 1.   {ECO:0000244|PubMed:22814378}; Name=p46Shc;   IsoId=P29353-3; Sequence=VSP_016107; Name=5;   IsoId=P29353-5; Sequence=VSP_040090, VSP_040091;   Note=Produced by alternative splicing.; Name=6;   IsoId=P29353-6; Sequence=VSP_040092;   Note=Produced by alternative splicing.; Name=7;   IsoId=P29353-7; Sequence=VSP_016108, VSP_040092;   Note=Produced by alternative splicing. Contains a   N-acetylmethionine at position 1. {ECO:0000244|PubMed:22814378};', 'TISSUE SPECIFICITY: Widely expressed. Expressed in neural stem cells but absent in mature neurons.', 'DOMAIN: In response to a variety of growth factors, isoform p46Shc and isoform p52Shc bind to phosphorylated Trk receptors through their phosphotyrosine binding (PID) and/or SH2 domains. The PID and SH2 domains bind to specific phosphorylated tyrosine residues in the Asn-Pro-Xaa-Tyr(P) motif of the Trk receptors. Isoform p46Shc and isoform p52Shc are in turn phosphorylated on three tyrosine residues within the extended proline-rich domain. These phosphotyrosines act as docking site for GRB2 and thereby are involved in Ras activation (By similarity). {ECO:0000250}.', 'PTM: Phosphorylated by activated epidermal growth factor receptor. Phosphorylated in response to FLT4 and KIT signaling. Isoform p46Shc and isoform p52Shc are phosphorylated on tyrosine residues of the Pro-rich domain. Isoform p66Shc is phosphorylated on Ser-36 by PRKCB upon treatment with insulin, hydrogen peroxide or irradiation with ultraviolet light (By similarity). Tyrosine phosphorylated in response to FLT3 signaling (By similarity). Tyrosine phosphorylated by activated PTK2B/PYK2 (By similarity). Tyrosine phosphorylated by ligand-activated ALK. Tyrosine phosphorylated by ligand-activated PDGFRB. Tyrosine phosphorylated by TEK/TIE2. May be tyrosine phosphorylated by activated PTK2/FAK1; tyrosine phosphorylation was seen in an astrocytoma biopsy, where PTK2/FAK1 kinase activity is high, but not in normal brain tissue. Isoform p52Shc dephosphorylation by PTPN2 may regulate interaction with GRB2. {ECO:0000250, ECO:0000269|PubMed:11980671, ECO:0000269|PubMed:14665640, ECO:0000269|PubMed:15526160, ECO:0000269|PubMed:15837797, ECO:0000269|PubMed:17274988, ECO:0000269|PubMed:8195171, ECO:0000269|PubMed:8939605, ECO:0000269|PubMed:9121430, ECO:0000269|PubMed:9148935, ECO:0000269|PubMed:9488479, ECO:0000269|PubMed:9566877}.', 'SIMILARITY: Contains 1 PID domain. {ECO:0000255|PROSITE- ProRule:PRU00148}.', 'SIMILARITY: Contains 1 SH2 domain. {ECO:0000255|PROSITE- ProRule:PRU00191}.', 'SEQUENCE CAUTION: Sequence=CAI13254.1; Type=Erroneous gene model prediction; Evidence={ECO:0000305};', 'WEB RESOURCE: Name=Atlas of Genetics and Cytogenetics in Oncology and Haematology; URL=\"http://atlasgeneticsoncology.org/Genes/SHC1ID42287ch1q21.html\";']\n",
      "created  ('01-DEC-1992', 0)\n",
      "cross_references  [('EMBL', 'X68148', 'CAA48251.1', '-', 'mRNA'), ('EMBL', 'U73377', 'AAB49972.1', '-', 'mRNA'), ('EMBL', 'Y09847', 'CAA70977.1', '-', 'Genomic_DNA'), ('EMBL', 'AK292143', 'BAF84832.1', '-', 'mRNA'), ('EMBL', 'AK315842', 'BAF98733.1', '-', 'mRNA'), ('EMBL', 'AB451255', 'BAG70069.1', '-', 'mRNA'), ('EMBL', 'AB451379', 'BAG70193.1', '-', 'mRNA'), ('EMBL', 'AL451085', 'CAI13248.1', '-', 'Genomic_DNA'), ('EMBL', 'AL451085', 'CAI13249.1', '-', 'Genomic_DNA'), ('EMBL', 'AL451085', 'CAI13250.1', '-', 'Genomic_DNA'), ('EMBL', 'AL451085', 'CAI13251.1', '-', 'Genomic_DNA'), ('EMBL', 'AL451085', 'CAI13254.1', 'ALT_SEQ', 'Genomic_DNA'), ('EMBL', 'CH471121', 'EAW53168.1', '-', 'Genomic_DNA'), ('EMBL', 'CH471121', 'EAW53169.1', '-', 'Genomic_DNA'), ('EMBL', 'CH471121', 'EAW53170.1', '-', 'Genomic_DNA'), ('EMBL', 'CH471121', 'EAW53171.1', '-', 'Genomic_DNA'), ('EMBL', 'BC014158', 'AAH14158.1', '-', 'mRNA'), ('EMBL', 'BC033925', 'AAH33925.1', '-', 'mRNA'), ('CCDS', 'CCDS1076.1', '-. [P29353-7]'), ('CCDS', 'CCDS30881.1', '-. [P29353-1]'), ('CCDS', 'CCDS44233.1', '-. [P29353-6]'), ('CCDS', 'CCDS44234.1', '-. [P29353-2]'), ('PIR', 'S25776', 'S25776'), ('RefSeq', 'NP_001123512.1', 'NM_001130040.1. [P29353-6]'), ('RefSeq', 'NP_001123513.1', 'NM_001130041.1. [P29353-2]'), ('RefSeq', 'NP_001189788.1', 'NM_001202859.1. [P29353-3]'), ('RefSeq', 'NP_003020.2', 'NM_003029.4. [P29353-7]'), ('RefSeq', 'NP_892113.4', 'NM_183001.4. [P29353-1]'), ('UniGene', 'Hs.433795', '-'), ('PDB', '1MIL', 'X-ray', '2.70 A', 'A=482-583'), ('PDB', '1N3H', 'NMR', '-', 'A=111-317'), ('PDB', '1OY2', 'NMR', '-', 'A=111-317'), ('PDB', '1QG1', 'NMR', '-', 'I=423-435'), ('PDB', '1SHC', 'NMR', '-', 'A=127-317'), ('PDB', '1TCE', 'NMR', '-', 'A=480-583'), ('PDB', '1WCP', 'Model', '-', 'A=127-583'), ('PDB', '2L1C', 'NMR', '-', 'A=127-317'), ('PDB', '4JMH', 'X-ray', '2.41 A', 'B=344-356'), ('PDB', '4XWX', 'X-ray', '1.87 A', 'A=147-311'), ('PDBsum', '1MIL', '-'), ('PDBsum', '1N3H', '-'), ('PDBsum', '1OY2', '-'), ('PDBsum', '1QG1', '-'), ('PDBsum', '1SHC', '-'), ('PDBsum', '1TCE', '-'), ('PDBsum', '1WCP', '-'), ('PDBsum', '2L1C', '-'), ('PDBsum', '4JMH', '-'), ('PDBsum', '4XWX', '-'), ('DisProt', 'DP00154', '-'), ('ProteinModelPortal', 'P29353', '-'), ('SMR', 'P29353', '111-317, 482-583'), ('BioGrid', '112361', '209'), ('DIP', 'DIP-699N', '-'), ('IntAct', 'P29353', '88'), ('MINT', 'MINT-123530', '-'), ('STRING', '9606.ENSP00000401303', '-'), ('BindingDB', 'P29353', '-'), ('ChEMBL', 'CHEMBL5626', '-'), ('PhosphoSite', 'P29353', '-'), ('BioMuta', 'SHC1', '-'), ('DMDM', '182676455', '-'), ('MaxQB', 'P29353', '-'), ('PaxDb', 'P29353', '-'), ('PRIDE', 'P29353', '-'), ('DNASU', '6464', '-'), ('Ensembl', 'ENST00000368445', 'ENSP00000357430', 'ENSG00000160691. [P29353-1]'), ('Ensembl', 'ENST00000368450', 'ENSP00000357435', 'ENSG00000160691. [P29353-2]'), ('Ensembl', 'ENST00000368453', 'ENSP00000357438', 'ENSG00000160691. [P29353-7]'), ('Ensembl', 'ENST00000448116', 'ENSP00000401303', 'ENSG00000160691. [P29353-6]'), ('GeneID', '6464', '-'), ('KEGG', 'hsa:6464', '-'), ('UCSC', 'uc001ffv.3', 'human. [P29353-1]'), ('UCSC', 'uc001ffw.3', 'human. [P29353-6]'), ('UCSC', 'uc001ffx.3', 'human. [P29353-7]'), ('CTD', '6464', '-'), ('GeneCards', 'SHC1', '-'), ('HGNC', 'HGNC:10840', 'SHC1'), ('HPA', 'CAB005374', '-'), ('HPA', 'CAB016305', '-'), ('HPA', 'HPA001844', '-'), ('MIM', '600560', 'gene'), ('neXtProt', 'NX_P29353', '-'), ('PharmGKB', 'PA35746', '-'), ('eggNOG', 'KOG3697', 'Eukaryota'), ('eggNOG', 'ENOG410XTJN', 'LUCA'), ('GeneTree', 'ENSGT00390000018860', '-'), ('HOVERGEN', 'HBG050121', '-'), ('InParanoid', 'P29353', '-'), ('KO', 'K06279', '-'), ('OMA', 'PRMSNLK', '-'), ('OrthoDB', 'EOG7MD4QK', '-'), ('PhylomeDB', 'P29353', '-'), ('TreeFam', 'TF315807', '-'), ('Reactome', 'R-HSA-114604', 'GPVI-mediated activation cascade'), ('Reactome', 'R-HSA-1236382', 'Constitutive Signaling by Ligand-Responsive EGFR Cancer Variants'), ('Reactome', 'R-HSA-1250196', 'SHC1 events in ERBB2 signaling'), ('Reactome', 'R-HSA-1250347', 'SHC1 events in ERBB4 signaling'), ('Reactome', 'R-HSA-167044', 'Signalling to RAS'), ('Reactome', 'R-HSA-180336', 'SHC1 events in EGFR signaling'), ('Reactome', 'R-HSA-210993', 'Tie2 Signaling'), ('Reactome', 'R-HSA-2424491', 'DAP12 signaling'), ('Reactome', 'R-HSA-2428933', 'SHC-related events triggered by IGF1R'), ('Reactome', 'R-HSA-2730905', 'Role of LAT2/NTAL/LAB on calcium mobilization'), ('Reactome', 'R-HSA-2871796', 'FCERI mediated MAPK activation'), ('Reactome', 'R-HSA-2871809', 'FCERI mediated Ca+2 mobilization'), ('Reactome', 'R-HSA-354192', 'Integrin alphaIIb beta3 signaling'), ('Reactome', 'R-HSA-381038', 'XBP1(S) activates chaperone genes'), ('Reactome', 'R-HSA-392451', 'G beta:gamma signalling through PI3Kgamma'), ('Reactome', 'R-HSA-451927', 'Interleukin-2 signaling'), ('Reactome', 'R-HSA-512988', 'Interleukin-3, 5 and GM-CSF signaling'), ('Reactome', 'R-HSA-5637810', 'Constitutive Signaling by EGFRvIII'), ('Reactome', 'R-HSA-5654688', 'SHC-mediated cascade:FGFR1'), ('Reactome', 'R-HSA-5654699', 'SHC-mediated cascade:FGFR2'), ('Reactome', 'R-HSA-5654704', 'SHC-mediated cascade:FGFR3'), ('Reactome', 'R-HSA-5654719', 'SHC-mediated cascade:FGFR4'), ('Reactome', 'R-HSA-5673001', 'RAF/MAP kinase cascade'), ('Reactome', 'R-HSA-74749', 'Signal attenuation'), ('Reactome', 'R-HSA-74751', 'Insulin receptor signalling cascade'), ('Reactome', 'R-HSA-912526', 'Interleukin receptor SHC signaling'), ('Reactome', 'R-HSA-983695', 'Antigen activates B Cell Receptor (BCR) leading to generation of second messengers'), ('SignaLink', 'P29353', '-'), ('ChiTaRS', 'SHC1', 'human'), ('EvolutionaryTrace', 'P29353', '-'), ('GeneWiki', 'SHC1', '-'), ('GenomeRNAi', '6464', '-'), ('NextBio', '25115', '-'), ('PRO', 'PR:P29353', '-'), ('Proteomes', 'UP000005640', 'Chromosome 1'), ('Bgee', 'P29353', '-'), ('ExpressionAtlas', 'P29353', 'baseline and differential'), ('Genevisible', 'P29353', 'HS'), ('GO', 'GO:0005829', 'C:cytosol', 'TAS:Reactome'), ('GO', 'GO:0010008', 'C:endosome membrane', 'IEA:Ensembl'), ('GO', 'GO:0005759', 'C:mitochondrial matrix', 'IEA:UniProtKB-SubCell'), ('GO', 'GO:0005634', 'C:nucleus', 'IEA:Ensembl'), ('GO', 'GO:0005886', 'C:plasma membrane', 'IDA:UniProtKB'), ('GO', 'GO:0070435', 'C:Shc-EGFR complex', 'ISS:BHF-UCL'), ('GO', 'GO:0046875', 'F:ephrin receptor binding', 'IPI:UniProtKB'), ('GO', 'GO:0005154', 'F:epidermal growth factor receptor binding', 'ISS:BHF-UCL'), ('GO', 'GO:0005158', 'F:insulin receptor binding', 'IPI:UniProtKB'), ('GO', 'GO:0005159', 'F:insulin-like growth factor receptor binding', 'IPI:UniProtKB'), ('GO', 'GO:0005168', 'F:neurotrophin TRKA receptor binding', 'IPI:UniProtKB'), ('GO', 'GO:0005543', 'F:phospholipid binding', 'TAS:UniProtKB'), ('GO', 'GO:0019901', 'F:protein kinase binding', 'IBA:GO_Central'), ('GO', 'GO:0004713', 'F:protein tyrosine kinase activity', 'TAS:Reactome'), ('GO', 'GO:0005068', 'F:transmembrane receptor protein tyrosine kinase adaptor activity', 'TAS:UniProtKB'), ('GO', 'GO:0031532', 'P:actin cytoskeleton reorganization', 'IEA:Ensembl'), ('GO', 'GO:0000187', 'P:activation of MAPK activity', 'IDA:UniProtKB'), ('GO', 'GO:0000186', 'P:activation of MAPKK activity', 'TAS:Reactome'), ('GO', 'GO:0007568', 'P:aging', 'IEA:Ensembl'), ('GO', 'GO:0001525', 'P:angiogenesis', 'IEA:UniProtKB-KW'), ('GO', 'GO:0007411', 'P:axon guidance', 'TAS:Reactome'), ('GO', 'GO:0007596', 'P:blood coagulation', 'TAS:Reactome'), ('GO', 'GO:0044267', 'P:cellular protein metabolic process', 'TAS:Reactome'), ('GO', 'GO:0071363', 'P:cellular response to growth factor stimulus', 'IDA:UniProtKB'), ('GO', 'GO:0030968', 'P:endoplasmic reticulum unfolded protein response', 'TAS:Reactome'), ('GO', 'GO:0007173', 'P:epidermal growth factor receptor signaling pathway', 'TAS:Reactome'), ('GO', 'GO:0038095', 'P:Fc-epsilon receptor signaling pathway', 'TAS:Reactome'), ('GO', 'GO:0008543', 'P:fibroblast growth factor receptor signaling pathway', 'TAS:Reactome'), ('GO', 'GO:0007507', 'P:heart development', 'IEA:Ensembl'), ('GO', 'GO:0045087', 'P:innate immune response', 'TAS:Reactome'), ('GO', 'GO:0008286', 'P:insulin receptor signaling pathway', 'ISS:BHF-UCL'), ('GO', 'GO:0036498', 'P:IRE1-mediated unfolded protein response', 'TAS:Reactome'), ('GO', 'GO:0050900', 'P:leukocyte migration', 'TAS:Reactome'), ('GO', 'GO:0000165', 'P:MAPK cascade', 'IDA:UniProtKB'), ('GO', 'GO:0048011', 'P:neurotrophin TRK receptor signaling pathway', 'TAS:Reactome'), ('GO', 'GO:0031100', 'P:organ regeneration', 'IEA:Ensembl'), ('GO', 'GO:0018108', 'P:peptidyl-tyrosine phosphorylation', 'TAS:GOC'), ('GO', 'GO:0030168', 'P:platelet activation', 'TAS:Reactome'), ('GO', 'GO:0008284', 'P:positive regulation of cell proliferation', 'NAS:UniProtKB'), ('GO', 'GO:0045740', 'P:positive regulation of DNA replication', 'ISS:BHF-UCL'), ('GO', 'GO:0048661', 'P:positive regulation of smooth muscle cell proliferation', 'IEA:Ensembl'), ('GO', 'GO:0045907', 'P:positive regulation of vasoconstriction', 'IEA:Ensembl'), ('GO', 'GO:0007265', 'P:Ras protein signal transduction', 'TAS:Reactome'), ('GO', 'GO:0007176', 'P:regulation of epidermal growth factor-activated receptor activity', 'TAS:ProtInc'), ('GO', 'GO:0040008', 'P:regulation of growth', 'IEA:UniProtKB-KW'), ('GO', 'GO:0051384', 'P:response to glucocorticoid', 'IEA:Ensembl'), ('GO', 'GO:0042542', 'P:response to hydrogen peroxide', 'IEA:Ensembl'), ('GO', 'GO:0001666', 'P:response to hypoxia', 'IEA:Ensembl'), ('GO', 'GO:0035094', 'P:response to nicotine', 'IEA:Ensembl'), ('GO', 'GO:0009636', 'P:response to toxic substance', 'IEA:Ensembl'), ('GO', 'GO:0016337', 'P:single organismal cell-cell adhesion', 'IEA:Ensembl'), ('GO', 'GO:0007264', 'P:small GTPase mediated signal transduction', 'TAS:Reactome'), ('GO', 'GO:0048010', 'P:vascular endothelial growth factor receptor signaling pathway', 'TAS:Reactome'), ('GO', 'GO:0016032', 'P:viral process', 'IEA:UniProtKB-KW'), ('Gene3D', '2.30.29.30', '-', '1'), ('Gene3D', '3.30.505.10', '-', '1'), ('InterPro', 'IPR011993', 'PH/PTB_dom'), ('InterPro', 'IPR006019', 'PID_Shc-like'), ('InterPro', 'IPR006020', 'PTB/PI_dom'), ('InterPro', 'IPR000980', 'SH2'), ('InterPro', 'IPR029586', 'Shc1/ShcA'), ('PANTHER', 'PTHR10337:SF2', 'PTHR10337:SF2', '1'), ('Pfam', 'PF00640', 'PID', '1'), ('Pfam', 'PF00017', 'SH2', '1'), ('PRINTS', 'PR00401', 'SH2DOMAIN'), ('PRINTS', 'PR00629', 'SHCPIDOMAIN'), ('SMART', 'SM00462', 'PTB', '1'), ('SMART', 'SM00252', 'SH2', '1'), ('SUPFAM', 'SSF55550', 'SSF55550', '1'), ('PROSITE', 'PS01179', 'PID', '1'), ('PROSITE', 'PS50001', 'SH2', '1')]\n",
      "data_class  Reviewed\n",
      "description  RecName: Full=SHC-transforming protein 1; AltName: Full=SHC-transforming protein 3; AltName: Full=SHC-transforming protein A; AltName: Full=Src homology 2 domain-containing-transforming protein C1; Short=SH2 domain protein C1;\n",
      "entry_name  SHC1_HUMAN\n",
      "features  [('CHAIN', 1, 583, 'SHC-transforming protein 1.', 'PRO_0000097731'), ('DOMAIN', 156, 339, 'PID. {ECO:0000255|PROSITE- ProRule:PRU00148}.', ''), ('DOMAIN', 488, 579, 'SH2. {ECO:0000255|PROSITE- ProRule:PRU00191}.', ''), ('REGION', 340, 487, 'CH1.', ''), ('COMPBIAS', 411, 474, 'Pro-rich.', ''), ('MOD_RES', 1, 1, 'N-acetylmethionine. {ECO:0000244|PubMed:22814378}.', ''), ('MOD_RES', 36, 36, 'Phosphoserine. {ECO:0000269|PubMed:15837797}.', ''), ('MOD_RES', 139, 139, 'Phosphoserine. {ECO:0000244|PubMed:17081983, ECO:0000244|PubMed:18669648, ECO:0000244|PubMed:21406692, ECO:0000244|PubMed:24275569}.', ''), ('MOD_RES', 154, 154, 'N6-acetyllysine. {ECO:0000250|UniProtKB:P98083}.', ''), ('MOD_RES', 349, 349, 'Phosphotyrosine. {ECO:0000269|PubMed:8939605, ECO:0000269|PubMed:9121430, ECO:0000269|PubMed:9566877}.', ''), ('MOD_RES', 350, 350, 'Phosphotyrosine. {ECO:0000269|PubMed:8939605, ECO:0000269|PubMed:9121430, ECO:0000269|PubMed:9566877}.', ''), ('MOD_RES', 427, 427, 'Phosphotyrosine. {ECO:0000244|PubMed:18669648, ECO:0000244|PubMed:19690332, ECO:0000269|PubMed:9121430, ECO:0000269|PubMed:9566877}.', ''), ('VAR_SEQ', 1, 214, 'Missing (in isoform 5). {ECO:0000305}.', 'VSP_040090'), ('VAR_SEQ', 1, 155, 'Missing (in isoform p46Shc). {ECO:0000303|PubMed:1623525}.', 'VSP_016107'), ('VAR_SEQ', 1, 110, 'Missing (in isoform p52Shc and isoform 7). {ECO:0000303|PubMed:14702039, ECO:0000303|PubMed:15489334, ECO:0000303|PubMed:1623525}.', 'VSP_016108'), ('VAR_SEQ', 215, 221, 'PLSSILG -> MSLCHRW (in isoform 5). {ECO:0000305}.', 'VSP_040091'), ('VAR_SEQ', 417, 417, 'P -> PA (in isoform 7 and isoform 6). {ECO:0000303|PubMed:15489334, ECO:0000303|PubMed:19054851}.', 'VSP_040092'), ('VARIANT', 205, 205, 'A -> V (in dbSNP:rs8191981).', 'VAR_042428'), ('VARIANT', 410, 410, 'M -> V (in dbSNP:rs8191979).', 'VAR_051353'), ('MUTAGEN', 349, 349, 'Y->F: Alters interaction with GRB2; isoform p52Shc (in vitro). {ECO:0000269|PubMed:9488479}.', ''), ('MUTAGEN', 427, 427, 'Y->F: No effect on interaction with GRB2; isoform p52Shc (in vitro). {ECO:0000269|PubMed:9488479}.', ''), ('CONFLICT', 2, 2, 'D -> N (in Ref. 3; CAA70977). {ECO:0000305}.', ''), ('CONFLICT', 21, 21, 'L -> M (in Ref. 3; CAA70977). {ECO:0000305}.', ''), ('CONFLICT', 38, 38, 'S -> P (in Ref. 3; CAA70977). {ECO:0000305}.', ''), ('CONFLICT', 95, 95, 'V -> D (in Ref. 2; AAB49972). {ECO:0000305}.', ''), ('CONFLICT', 101, 101, 'D -> E (in Ref. 2; AAB49972). {ECO:0000305}.', ''), ('CONFLICT', 430, 430, 'V -> A (in Ref. 5; BAG70069/BAG70193). {ECO:0000305}.', ''), ('STRAND', 113, 115, '{ECO:0000244|PDB:1N3H}.', ''), ('STRAND', 142, 145, '{ECO:0000244|PDB:1SHC}.', ''), ('STRAND', 148, 150, '{ECO:0000244|PDB:1N3H}.', ''), ('HELIX', 152, 156, '{ECO:0000244|PDB:4XWX}.', ''), ('STRAND', 160, 172, '{ECO:0000244|PDB:4XWX}.', ''), ('HELIX', 176, 178, '{ECO:0000244|PDB:4XWX}.', ''), ('HELIX', 181, 198, '{ECO:0000244|PDB:4XWX}.', ''), ('HELIX', 200, 202, '{ECO:0000244|PDB:1OY2}.', ''), ('STRAND', 206, 208, '{ECO:0000244|PDB:1SHC}.', ''), ('TURN', 215, 219, '{ECO:0000244|PDB:4XWX}.', ''), ('STRAND', 220, 224, '{ECO:0000244|PDB:4XWX}.', ''), ('TURN', 226, 229, '{ECO:0000244|PDB:4XWX}.', ''), ('STRAND', 230, 236, '{ECO:0000244|PDB:4XWX}.', ''), ('STRAND', 238, 245, '{ECO:0000244|PDB:4XWX}.', ''), ('TURN', 246, 249, '{ECO:0000244|PDB:1N3H}.', ''), ('STRAND', 251, 256, '{ECO:0000244|PDB:4XWX}.', ''), ('HELIX', 257, 259, '{ECO:0000244|PDB:4XWX}.', ''), ('STRAND', 262, 265, '{ECO:0000244|PDB:4XWX}.', ''), ('HELIX', 268, 270, '{ECO:0000244|PDB:4XWX}.', ''), ('STRAND', 273, 280, '{ECO:0000244|PDB:4XWX}.', ''), ('TURN', 281, 283, '{ECO:0000244|PDB:4XWX}.', ''), ('STRAND', 284, 291, '{ECO:0000244|PDB:4XWX}.', ''), ('STRAND', 293, 295, '{ECO:0000244|PDB:1N3H}.', ''), ('HELIX', 296, 310, '{ECO:0000244|PDB:4XWX}.', ''), ('STRAND', 311, 314, '{ECO:0000244|PDB:1N3H}.', ''), ('TURN', 483, 485, '{ECO:0000244|PDB:1MIL}.', ''), ('STRAND', 487, 489, '{ECO:0000244|PDB:1TCE}.', ''), ('HELIX', 495, 499, '{ECO:0000244|PDB:1MIL}.', ''), ('STRAND', 507, 512, '{ECO:0000244|PDB:1MIL}.', ''), ('STRAND', 514, 516, '{ECO:0000244|PDB:1TCE}.', ''), ('STRAND', 518, 526, '{ECO:0000244|PDB:1MIL}.', ''), ('STRAND', 529, 536, '{ECO:0000244|PDB:1MIL}.', ''), ('STRAND', 540, 543, '{ECO:0000244|PDB:1MIL}.', ''), ('STRAND', 548, 551, '{ECO:0000244|PDB:1MIL}.', ''), ('HELIX', 552, 562, '{ECO:0000244|PDB:1MIL}.', ''), ('STRAND', 566, 568, '{ECO:0000244|PDB:1MIL}.', ''), ('STRAND', 571, 573, '{ECO:0000244|PDB:1MIL}.', '')]\n",
      "gene_name  Name=SHC1; Synonyms=SHC, SHCA;\n",
      "host_organism  []\n",
      "host_taxonomy_id  []\n",
      "keywords  ['3D-structure', 'Acetylation', 'Alternative promoter usage', 'Alternative splicing', 'Angiogenesis', 'Complete proteome', 'Cytoplasm', 'Growth regulation', 'Host-virus interaction', 'Mitochondrion', 'Phosphoprotein', 'Polymorphism', 'Reference proteome', 'SH2 domain']\n",
      "molecule_type  None\n",
      "organelle  \n",
      "organism  Homo sapiens (Human).\n",
      "organism_classification  ['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Mammalia', 'Eutheria', 'Euarchontoglires', 'Primates', 'Haplorrhini', 'Catarrhini', 'Hominidae', 'Homo']\n",
      "references  [<Bio.SwissProt.Reference object at 0x1067d8898>, <Bio.SwissProt.Reference object at 0x1067d8710>, <Bio.SwissProt.Reference object at 0x1067d8978>, <Bio.SwissProt.Reference object at 0x1067d8b38>, <Bio.SwissProt.Reference object at 0x1067d8f28>, <Bio.SwissProt.Reference object at 0x1067d5080>, <Bio.SwissProt.Reference object at 0x1067d8940>, <Bio.SwissProt.Reference object at 0x1067d5208>, <Bio.SwissProt.Reference object at 0x1067d51d0>, <Bio.SwissProt.Reference object at 0x1067d52e8>, <Bio.SwissProt.Reference object at 0x1067d5240>, <Bio.SwissProt.Reference object at 0x1067d53c8>, <Bio.SwissProt.Reference object at 0x1067d5588>, <Bio.SwissProt.Reference object at 0x1067d5630>, <Bio.SwissProt.Reference object at 0x1067d56d8>, <Bio.SwissProt.Reference object at 0x1067f3278>, <Bio.SwissProt.Reference object at 0x1067d5550>, <Bio.SwissProt.Reference object at 0x1067f3e10>, <Bio.SwissProt.Reference object at 0x1067f3dd8>, <Bio.SwissProt.Reference object at 0x1067f3fd0>, <Bio.SwissProt.Reference object at 0x1067f32e8>, <Bio.SwissProt.Reference object at 0x1067f3320>, <Bio.SwissProt.Reference object at 0x1067b50f0>, <Bio.SwissProt.Reference object at 0x1067b5da0>, <Bio.SwissProt.Reference object at 0x1067b5a58>, <Bio.SwissProt.Reference object at 0x1067b52b0>, <Bio.SwissProt.Reference object at 0x106802278>, <Bio.SwissProt.Reference object at 0x1067b5cc0>, <Bio.SwissProt.Reference object at 0x106802e10>, <Bio.SwissProt.Reference object at 0x106802e48>, <Bio.SwissProt.Reference object at 0x106802eb8>, <Bio.SwissProt.Reference object at 0x106802f28>, <Bio.SwissProt.Reference object at 0x106802240>, <Bio.SwissProt.Reference object at 0x106802da0>, <Bio.SwissProt.Reference object at 0x1067e18d0>, <Bio.SwissProt.Reference object at 0x1067e17f0>, <Bio.SwissProt.Reference object at 0x1067e19e8>, <Bio.SwissProt.Reference object at 0x1067e1a58>, <Bio.SwissProt.Reference object at 0x1067e1a90>, <Bio.SwissProt.Reference object at 0x1067e1ba8>, <Bio.SwissProt.Reference object at 0x1067e1cc0>, <Bio.SwissProt.Reference object at 0x1067e1d68>, <Bio.SwissProt.Reference object at 0x1067e1e10>, <Bio.SwissProt.Reference object at 0x1067e1f60>, <Bio.SwissProt.Reference object at 0x1067e1ef0>, <Bio.SwissProt.Reference object at 0x1067e1f98>, <Bio.SwissProt.Reference object at 0x1067ad128>, <Bio.SwissProt.Reference object at 0x1067ad240>, <Bio.SwissProt.Reference object at 0x1067ad2b0>, <Bio.SwissProt.Reference object at 0x1067d8358>, <Bio.SwissProt.Reference object at 0x1068023c8>, <Bio.SwissProt.Reference object at 0x106802550>, <Bio.SwissProt.Reference object at 0x106802400>, <Bio.SwissProt.Reference object at 0x106802470>]\n",
      "seqinfo  (583, 62822, '7EFA5CB185A548D1')\n",
      "sequence  MDLLPPKPKYNPLRNESLSSLEEGASGSTPPEELPSPSASSLGPILPPLPGDDSPTTLCSFFPRMSNLRLANPAGGRPGSKGEPGRAADDGEGIVGAAMPDSGPLPLLQDMNKLSGGGGRRTRVEGGQLGGEEWTRHGSFVNKPTRGWLHPNDKVMGPGVSYLVRYMGCVEVLQSMRALDFNTRTQVTREAISLVCEAVPGAKGATRRRKPCSRPLSSILGRSNLKFAGMPITLTVSTSSLNLMAADCKQIIANHHMQSISFASGGDPDTAEYVAYVAKDPVNQRACHILECPEGLAQDVISTIGQAFELRFKQYLRNPPKLVTPHDRMAGFDGSAWDEEEEEPPDHQYYNDFPGKEPPLGGVVDMRLREGAAPGAARPTAPNAQTPSHLGATLPVGQPVGGDPEVRKQMPPPPPCPGRELFDDPSYVNVQNLDKARQAVGGAGPPNPAINGSAPRDLFDMKPFEDALRVPPPPQSVSMAEQLRGEPWFHGKLSRREAEALLQLNGDFLVRESTTTPGQYVLTGLQSGQPKHLLLVDPEGVVRTKDHRFESVSHLISYHMDNHLPIISAGSELCLQQPVERKL\n",
      "sequence_length  583\n",
      "sequence_update  ('08-APR-2008', 0)\n",
      "taxonomy_id  ['9606']\n"
     ]
    }
   ],
   "source": [
    "from Bio import ExPASy, SwissProt\n",
    "handle = ExPASy.get_sprot_raw(\"P29353\")\n",
    "\n",
    "record = SwissProt.read(handle)\n",
    "print(dir(record))\n",
    "print(\"accessions \", record.accessions)\n",
    "print(\"annotation_update \",record.annotation_update)\n",
    "print(\"comments \",record.comments)\n",
    "print(\"created \", record.created)\n",
    "print(\"cross_references \",record.cross_references)\n",
    "print(\"data_class \",record.data_class)\n",
    "print(\"description \",record.description)\n",
    "print(\"entry_name \",record.entry_name)\n",
    "print(\"features \", record.features)\n",
    "print(\"gene_name \",record.gene_name)\n",
    "print(\"host_organism \",record.host_organism)\n",
    "print(\"host_taxonomy_id \",record.host_taxonomy_id)\n",
    "print(\"keywords \",record.keywords)\n",
    "print(\"molecule_type \",record.molecule_type)\n",
    "print(\"organelle \",record.organelle)\n",
    "print(\"organism \",record.organism)\n",
    "print(\"organism_classification \",record.organism_classification)\n",
    "print(\"references \", record.references)\n",
    "print(\"seqinfo \",record.seqinfo)\n",
    "print(\"sequence \",record.sequence)\n",
    "print(\"sequence_length \",record.sequence_length)\n",
    "print(\"sequence_update \",record.sequence_update)\n",
    "print(\"taxonomy_id \",record.taxonomy_id)\n",
    "\n",
    "# for attribute in dir(record):\n",
    "#     print(record[\"attribute\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "authors  Pelicci G., Lanfrancone L., Grignani F., McGlade J., Cavallo F., Forni G., Nicoletti I., Grignani F., Pawson T., Pelicci P.-G.\n",
      "comments  []\n",
      "location  Cell 70:93-104(1992).\n",
      "number  1\n",
      "positions  ['NUCLEOTIDE SEQUENCE [MRNA] (ISOFORMS P46SHC AND P52SHC).']\n",
      "references  [('PubMed', '1623525'), ('DOI', '10.1016/0092-8674(92)90536-L')]\n",
      "title  A novel transforming protein (SHC) with an SH2 domain is implicated in mitogenic signal transduction.\n",
      "authors  Migliaccio E., Mele S., Salcini A.E., Pelicci G., Lai K.M., Superti-Furga G., Pawson T., Di Fiore P.P., Lanfrancone L., Pelicci P.-G.\n",
      "comments  []\n",
      "location  EMBO J. 16:706-716(1997).\n",
      "number  2\n",
      "positions  ['NUCLEOTIDE SEQUENCE [MRNA] (ISOFORM P66SHC).']\n",
      "references  [('PubMed', '9049300'), ('DOI', '10.1093/emboj/16.4.706')]\n",
      "title  Opposite effects of the p52shc/p46shc and p66shc splicing isoforms on the EGF receptor-MAP kinase-fos signalling pathway.\n",
      "authors  Harun R.B., Smith K.K., Leek J.P., Markham A.F., Norris A., Morrison J.F.\n",
      "comments  [('TISSUE', 'Fibroblast')]\n",
      "location  Genomics 42:349-352(1997).\n",
      "number  3\n",
      "positions  ['NUCLEOTIDE SEQUENCE [GENOMIC DNA].']\n",
      "references  [('PubMed', '9192859'), ('DOI', '10.1006/geno.1997.4728')]\n",
      "title  Characterization of human SHC p66 cDNA and its processed pseudogene mapping to Xq12-q13.1.\n",
      "authors  Ota T., Suzuki Y., Nishikawa T., Otsuki T., Sugiyama T., Irie R., Wakamatsu A., Hayashi K., Sato H., Nagai K., Kimura K., Makita H., Sekine M., Obayashi M., Nishi T., Shibahara T., Tanaka T., Ishii S., Yamamoto J., Saito K., Kawai Y., Isono Y., Nakamura Y., Nagahari K., Murakami K., Yasuda T., Iwayanagi T., Wagatsuma M., Shiratori A., Sudo H., Hosoiri T., Kaku Y., Kodaira H., Kondo H., Sugawara M., Takahashi M., Kanda K., Yokoi T., Furuya T., Kikkawa E., Omura Y., Abe K., Kamihara K., Katsuta N., Sato K., Tanikawa M., Yamazaki M., Ninomiya K., Ishibashi T., Yamashita H., Murakawa K., Fujimori K., Tanai H., Kimata M., Watanabe M., Hiraoka S., Chiba Y., Ishida S., Ono Y., Takiguchi S., Watanabe S., Yosida M., Hotuta T., Kusano J., Kanehori K., Takahashi-Fujii A., Hara H., Tanase T.-O., Nomura Y., Togiya S., Komai F., Hara R., Takeuchi K., Arita M., Imose N., Musashino K., Yuuki H., Oshima A., Sasaki N., Aotsuka S., Yoshikawa Y., Matsunawa H., Ichihara T., Shiohata N., Sano S., Moriya S., Momiyama H., Satoh N., Takami S., Terashima Y., Suzuki O., Nakagawa S., Senoh A., Mizoguchi H., Goto Y., Shimizu F., Wakebe H., Hishigaki H., Watanabe T., Sugiyama A., Takemoto M., Kawakami B., Yamazaki M., Watanabe K., Kumagai A., Itakura S., Fukuzumi Y., Fujimori Y., Komiyama M., Tashiro H., Tanigami A., Fujiwara T., Ono T., Yamada K., Fujii Y., Ozaki K., Hirao M., Ohmori Y., Kawabata A., Hikiji T., Kobatake N., Inagaki H., Ikema Y., Okamoto S., Okitani R., Kawakami T., Noguchi S., Itoh T., Shigeta K., Senba T., Matsumura K., Nakajima Y., Mizuno T., Morinaga M., Sasaki M., Togashi T., Oyama M., Hata H., Watanabe M., Komatsu T., Mizushima-Sugano J., Satoh T., Shirai Y., Takahashi Y., Nakagawa K., Okumura K., Nagase T., Nomura N., Kikuchi H., Masuho Y., Yamashita R., Nakai K., Yada T., Nakamura Y., Ohara O., Isogai T., Sugano S.\n",
      "comments  [('TISSUE', 'Mammary gland, and Synovium')]\n",
      "location  Nat. Genet. 36:40-45(2004).\n",
      "number  4\n",
      "positions  ['NUCLEOTIDE SEQUENCE [LARGE SCALE MRNA] (ISOFORM P52SHC).']\n",
      "references  [('PubMed', '14702039'), ('DOI', '10.1038/ng1285')]\n",
      "title  Complete sequencing and characterization of 21,243 full-length human cDNAs.\n",
      "authors  Goshima N., Kawamura Y., Fukumoto A., Miura A., Honma R., Satoh R., Wakamatsu A., Yamamoto J., Kimura K., Nishikawa T., Andoh T., Iida Y., Ishikawa K., Ito E., Kagawa N., Kaminaga C., Kanehori K., Kawakami B., Kenmochi K., Kimura R., Kobayashi M., Kuroita T., Kuwayama H., Maruyama Y., Matsuo K., Minami K., Mitsubori M., Mori M., Morishita R., Murase A., Nishikawa A., Nishikawa S., Okamoto T., Sakagami N., Sakamoto Y., Sasaki Y., Seki T., Sono S., Sugiyama A., Sumiya T., Takayama T., Takayama Y., Takeda H., Togashi T., Yahata K., Yamada H., Yanagisawa Y., Endo Y., Imamoto F., Kisu Y., Tanaka S., Isogai T., Imai J., Watanabe S., Nomura N.\n",
      "comments  []\n",
      "location  Nat. Methods 5:1011-1017(2008).\n",
      "number  5\n",
      "positions  ['NUCLEOTIDE SEQUENCE [LARGE SCALE MRNA] (ISOFORM 6).']\n",
      "references  [('PubMed', '19054851'), ('DOI', '10.1038/nmeth.1273')]\n",
      "title  Human protein factory for converting the transcriptome into an in vitro-expressed proteome.\n",
      "authors  Gregory S.G., Barlow K.F., McLay K.E., Kaul R., Swarbreck D., Dunham A., Scott C.E., Howe K.L., Woodfine K., Spencer C.C.A., Jones M.C., Gillson C., Searle S., Zhou Y., Kokocinski F., McDonald L., Evans R., Phillips K., Atkinson A., Cooper R., Jones C., Hall R.E., Andrews T.D., Lloyd C., Ainscough R., Almeida J.P., Ambrose K.D., Anderson F., Andrew R.W., Ashwell R.I.S., Aubin K., Babbage A.K., Bagguley C.L., Bailey J., Beasley H., Bethel G., Bird C.P., Bray-Allen S., Brown J.Y., Brown A.J., Buckley D., Burton J., Bye J., Carder C., Chapman J.C., Clark S.Y., Clarke G., Clee C., Cobley V., Collier R.E., Corby N., Coville G.J., Davies J., Deadman R., Dunn M., Earthrowl M., Ellington A.G., Errington H., Frankish A., Frankland J., French L., Garner P., Garnett J., Gay L., Ghori M.R.J., Gibson R., Gilby L.M., Gillett W., Glithero R.J., Grafham D.V., Griffiths C., Griffiths-Jones S., Grocock R., Hammond S., Harrison E.S.I., Hart E., Haugen E., Heath P.D., Holmes S., Holt K., Howden P.J., Hunt A.R., Hunt S.E., Hunter G., Isherwood J., James R., Johnson C., Johnson D., Joy A., Kay M., Kershaw J.K., Kibukawa M., Kimberley A.M., King A., Knights A.J., Lad H., Laird G., Lawlor S., Leongamornlert D.A., Lloyd D.M., Loveland J., Lovell J., Lush M.J., Lyne R., Martin S., Mashreghi-Mohammadi M., Matthews L., Matthews N.S.W., McLaren S., Milne S., Mistry S., Moore M.J.F., Nickerson T., O'Dell C.N., Oliver K., Palmeiri A., Palmer S.A., Parker A., Patel D., Pearce A.V., Peck A.I., Pelan S., Phelps K., Phillimore B.J., Plumb R., Rajan J., Raymond C., Rouse G., Saenphimmachak C., Sehra H.K., Sheridan E., Shownkeen R., Sims S., Skuce C.D., Smith M., Steward C., Subramanian S., Sycamore N., Tracey A., Tromans A., Van Helmond Z., Wall M., Wallis J.M., White S., Whitehead S.L., Wilkinson J.E., Willey D.L., Williams H., Wilming L., Wray P.W., Wu Z., Coulson A., Vaudin M., Sulston J.E., Durbin R.M., Hubbard T., Wooster R., Dunham I., Carter N.P., McVean G., Ross M.T., Harrow J., Olson M.V., Beck S., Rogers J., Bentley D.R.\n",
      "comments  []\n",
      "location  Nature 441:315-321(2006).\n",
      "number  6\n",
      "positions  ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA].']\n",
      "references  [('PubMed', '16710414'), ('DOI', '10.1038/nature04727')]\n",
      "title  The DNA sequence and biological annotation of human chromosome 1.\n",
      "authors  Mural R.J., Istrail S., Sutton G.G., Florea L., Halpern A.L., Mobarry C.M., Lippert R., Walenz B., Shatkay H., Dew I., Miller J.R., Flanigan M.J., Edwards N.J., Bolanos R., Fasulo D., Halldorsson B.V., Hannenhalli S., Turner R., Yooseph S., Lu F., Nusskern D.R., Shue B.C., Zheng X.H., Zhong F., Delcher A.L., Huson D.H., Kravitz S.A., Mouchard L., Reinert K., Remington K.A., Clark A.G., Waterman M.S., Eichler E.E., Adams M.D., Hunkapiller M.W., Myers E.W., Venter J.C.\n",
      "comments  []\n",
      "location  Submitted (SEP-2005) to the EMBL/GenBank/DDBJ databases.\n",
      "number  7\n",
      "positions  ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA].']\n",
      "references  []\n",
      "title  \n",
      "authors  The MGC Project Team\n",
      "comments  [('TISSUE', 'Choriocarcinoma, and Neuroblastoma')]\n",
      "location  Genome Res. 14:2121-2127(2004).\n",
      "number  8\n",
      "positions  ['NUCLEOTIDE SEQUENCE [LARGE SCALE MRNA] (ISOFORMS P52SHC AND 7).']\n",
      "references  [('PubMed', '15489334'), ('DOI', '10.1101/gr.2596504')]\n",
      "title  The status, quality, and expansion of the NIH full-length cDNA project: the Mammalian Gene Collection (MGC).\n",
      "authors  Yokote K., Mori S., Hansen K., McGlade J., Pawson T., Heldin C.H., Claesson-Welsh L.\n",
      "comments  []\n",
      "location  J. Biol. Chem. 269:15337-15343(1994).\n",
      "number  9\n",
      "positions  ['INTERACTION WITH PDGFRB AND GRB2, AND PHOSPHORYLATION.']\n",
      "references  [('PubMed', '8195171')]\n",
      "title  Direct interaction between Shc and the platelet-derived growth factor beta-receptor.\n",
      "authors  Stephens R.M., Loeb D.M., Copeland T.D., Pawson T., Greene L.A., Kaplan D.R.\n",
      "comments  []\n",
      "location  Neuron 12:691-705(1994).\n",
      "number  10\n",
      "positions  ['INTERACTION WITH NTRK1.']\n",
      "references  [('PubMed', '8155326'), ('DOI', '10.1016/0896-6273(94)90223-2')]\n",
      "title  Trk receptors use redundant signal transduction pathways involving SHC and PLC-gamma 1 to mediate NGF responses.\n",
      "authors  Craparo A., O'Neill T.J., Gustafson T.A.\n",
      "comments  []\n",
      "location  J. Biol. Chem. 270:15639-15643(1995).\n",
      "number  11\n",
      "positions  ['INTERACTION WITH IGF1R.']\n",
      "references  [('PubMed', '7541045'), ('DOI', '10.1074/jbc.270.26.15639')]\n",
      "title  Non-SH2 domains within insulin receptor substrate-1 and SHC mediate their phosphotyrosine-dependent interaction with the NPEY motif of the insulin-like growth factor I receptor.\n",
      "authors  He W., O'Neill T.J., Gustafson T.A.\n",
      "comments  []\n",
      "location  J. Biol. Chem. 270:23258-23262(1995).\n",
      "number  12\n",
      "positions  ['INTERACTION WITH INSR.']\n",
      "references  [('PubMed', '7559478'), ('DOI', '10.1074/jbc.270.40.23258')]\n",
      "title  Distinct modes of interaction of SHC and insulin receptor substrate-1 with the insulin receptor NPEY region via non-SH2 domains.\n",
      "authors  Gustafson T.A., He W., Craparo A., Schaub C.D., O'Neill T.J.\n",
      "comments  []\n",
      "location  Mol. Cell. Biol. 15:2500-2508(1995).\n",
      "number  13\n",
      "positions  ['INTERACTION WITH INSR.']\n",
      "references  [('PubMed', '7537849')]\n",
      "title  Phosphotyrosine-dependent interaction of SHC and insulin receptor substrate 1 with the NPEY motif of the insulin receptor via a novel non-SH2 domain.\n",
      "authors  Ware M.D., Rosten P., Damen J.E., Liu L., Humphries R.K., Krystal G.\n",
      "comments  []\n",
      "location  Blood 88:2833-2840(1996).\n",
      "number  14\n",
      "positions  ['INTERACTION WITH INPP5D.']\n",
      "references  [('PubMed', '8874179')]\n",
      "title  Cloning and characterization of human SHIP, the 145-kD inositol 5- phosphatase that associates with SHC after cytokine stimulation.\n",
      "authors  van der Geer P., Wiley S., Gish G.D., Pawson T.\n",
      "comments  []\n",
      "location  Curr. Biol. 6:1435-1444(1996).\n",
      "number  15\n",
      "positions  ['PHOSPHORYLATION AT TYR-349 AND TYR-350.']\n",
      "references  [('PubMed', '8939605'), ('DOI', '10.1016/S0960-9822(96)00748-8')]\n",
      "title  The Shc adaptor protein is highly phosphorylated at conserved, twin tyrosine residues (Y239/240) that mediate protein-protein interactions.\n",
      "authors  Yokote K., Margolis B., Heldin C.H., Claesson-Welsh L.\n",
      "comments  []\n",
      "location  J. Biol. Chem. 271:30942-30949(1996).\n",
      "number  16\n",
      "positions  ['INTERACTION WITH GRB7.']\n",
      "references  [('PubMed', '8940081'), ('DOI', '10.1074/jbc.271.48.30942')]\n",
      "title  Grb7 is a downstream signaling component of platelet-derived growth factor alpha- and beta-receptors.\n",
      "authors  Price D.J., Rivnay B., Fu Y., Jiang S., Avraham S., Avraham H.\n",
      "comments  []\n",
      "location  J. Biol. Chem. 272:5915-5920(1997).\n",
      "number  17\n",
      "positions  ['INTERACTION WITH KIT.']\n",
      "references  [('PubMed', '9038210'), ('DOI', '10.1074/jbc.272.9.5915')]\n",
      "title  Direct association of Csk homologous kinase (CHK) with the diphosphorylated site Tyr568/570 of the activated c-KIT in megakaryocytes.\n",
      "authors  Schlaepfer D.D., Hunter T.\n",
      "comments  []\n",
      "location  J. Biol. Chem. 272:13189-13195(1997).\n",
      "number  18\n",
      "positions  ['PHOSPHORYLATION.']\n",
      "references  [('PubMed', '9148935'), ('DOI', '10.1074/jbc.272.20.13189')]\n",
      "title  Focal adhesion kinase overexpression enhances ras-dependent integrin signaling to ERK2/mitogen-activated protein kinase through interactions with and activation of c-Src.\n",
      "authors  Gotoh N., Toyoda M., Shibuya M.\n",
      "comments  []\n",
      "location  Mol. Cell. Biol. 17:1824-1831(1997).\n",
      "number  19\n",
      "positions  ['PHOSPHORYLATION AT TYR-349; TYR-350 AND TYR-427.']\n",
      "references  [('PubMed', '9121430')]\n",
      "title  Tyrosine phosphorylation sites at amino acids 239 and 240 of Shc are involved in epidermal growth factor-induced mitogenic signaling that is distinct from Ras/mitogen-activated protein kinase activation.\n",
      "authors  Yokouchi M., Suzuki R., Masuhara M., Komiya S., Inoue A., Yoshimura A.\n",
      "comments  []\n",
      "location  Oncogene 15:7-15(1997).\n",
      "number  20\n",
      "positions  ['INTERACTION WITH APS.']\n",
      "references  [('PubMed', '9233773'), ('DOI', '10.1038/sj.onc.1201163')]\n",
      "title  Cloning and characterization of APS, an adaptor molecule containing PH and SH2 domains that is tyrosine phosphorylated upon B-cell receptor stimulation.\n",
      "authors  Habib T., Hejna J.A., Moses R.E., Decker S.J.\n",
      "comments  []\n",
      "location  J. Biol. Chem. 273:18605-18609(1998).\n",
      "number  21\n",
      "positions  ['INTERACTION WITH INPP5D AND INPPL1.']\n",
      "references  [('PubMed', '9660833'), ('DOI', '10.1074/jbc.273.29.18605')]\n",
      "title  Growth factors and insulin stimulate tyrosine phosphorylation of the 51C/SHIP2 protein.\n",
      "authors  Tiganis T., Bennett A.M., Ravichandran K.S., Tonks N.K.\n",
      "comments  []\n",
      "location  Mol. Cell. Biol. 18:1622-1634(1998).\n",
      "number  22\n",
      "positions  ['PHOSPHORYLATION, DEPHOSPHORYLATION BY PTPN2, MUTAGENESIS OF TYR-349', 'AND TYR-427, AND INTERACTION WITH GRB2.']\n",
      "references  [('PubMed', '9488479')]\n",
      "title  Epidermal growth factor receptor and the adaptor protein p52Shc are specific substrates of T-cell protein tyrosine phosphatase.\n",
      "authors  Schlaepfer D.D., Jones K.C., Hunter T.\n",
      "comments  []\n",
      "location  Mol. Cell. Biol. 18:2571-2585(1998).\n",
      "number  23\n",
      "positions  ['PHOSPHORYLATION AT TYR-349; TYR-350 AND TYR-427.']\n",
      "references  [('PubMed', '9566877')]\n",
      "title  Multiple Grb2-mediated integrin-stimulated signaling pathways to ERK2/mitogen-activated protein kinase: summation of both c-Src- and focal adhesion kinase-initiated tyrosine phosphorylation events.\n",
      "authors  Wisniewski D., Strife A., Swendeman S., Erdjument-Bromage H., Geromanos S., Kavanaugh W.M., Tempst P., Clarkson B.\n",
      "comments  []\n",
      "location  Blood 93:2707-2720(1999).\n",
      "number  24\n",
      "positions  ['INTERACTION WITH INPPL1.']\n",
      "references  [('PubMed', '10194451')]\n",
      "title  A novel SH2-containing phosphatidylinositol 3,4,5-trisphosphate 5- phosphatase (SHIP2) is constitutively tyrosine phosphorylated and associated with src homologous and collagen gene (SHC) in chronic myelogenous leukemia progenitor cells.\n",
      "authors  Sweeney C., Lai C., Riese D.J. II, Diamonti A.J., Cantley L.C., Carraway K.L. III\n",
      "comments  []\n",
      "location  J. Biol. Chem. 275:19803-19807(2000).\n",
      "number  25\n",
      "positions  ['INTERACTION WITH ERBB4.']\n",
      "references  [('PubMed', '10867024'), ('DOI', '10.1074/jbc.C901015199')]\n",
      "title  Ligand discrimination in signaling through an ErbB4 receptor homodimer.\n",
      "authors  Pesesse X., Dewaste V., De Smedt F., Laffargue M., Giuriato S., Moreau C., Payrastre B., Erneux C.\n",
      "comments  []\n",
      "location  J. Biol. Chem. 276:28348-28355(2001).\n",
      "number  26\n",
      "positions  ['INTERACTION WITH INPPL1.']\n",
      "references  [('PubMed', '11349134'), ('DOI', '10.1074/jbc.M103537200')]\n",
      "title  The Src homology 2 domain containing inositol 5-phosphatase SHIP2 is recruited to the epidermal growth factor (EGF) receptor and dephosphorylates phosphatidylinositol 3,4,5-trisphosphate in EGF- stimulated COS-7 cells.\n",
      "authors  Hecker T.P., Grammer J.R., Gillespie G.Y., Stewart J. Jr., Gladson C.L.\n",
      "comments  []\n",
      "location  Cancer Res. 62:2699-2707(2002).\n",
      "number  27\n",
      "positions  ['INTERACTION WITH PTK2/FAK1, AND PHOSPHORYLATION.']\n",
      "references  [('PubMed', '11980671')]\n",
      "title  Focal adhesion kinase enhances signaling through the Shc/extracellular signal-regulated kinase pathway in anaplastic astrocytoma tumor biopsy samples.\n",
      "authors  Ventura A., Luzi L., Pacini S., Baldari C.T., Pelicci P.-G.\n",
      "comments  []\n",
      "location  J. Biol. Chem. 277:22370-22376(2002).\n",
      "number  28\n",
      "positions  ['ALTERNATIVE PROMOTER USAGE.']\n",
      "references  [('PubMed', '11948181'), ('DOI', '10.1074/jbc.M200280200')]\n",
      "title  The p66Shc longevity gene is silenced through epigenetic modifications of an alternative promoter.\n",
      "authors  Vindis C., Cerretti D.P., Daniel T.O., Huynh-Do U.\n",
      "comments  []\n",
      "location  J. Cell Biol. 162:661-671(2003).\n",
      "number  29\n",
      "positions  ['INTERACTION WITH EPHB1 AND GRB2.']\n",
      "references  [('PubMed', '12925710'), ('DOI', '10.1083/jcb.200302073')]\n",
      "title  EphB1 recruits c-Src and p52Shc to activate MAPK/ERK and promote chemotaxis.\n",
      "authors  Tacconelli A., Farina A.R., Cappabianca L., Desantis G., Tessitore A., Vetuschi A., Sferra R., Rucci N., Argenti B., Screpanti I., Gulino A., Mackay A.R.\n",
      "comments  []\n",
      "location  Cancer Cell 6:347-360(2004).\n",
      "number  30\n",
      "positions  ['INTERACTION WITH NTRK1.']\n",
      "references  [('PubMed', '15488758'), ('DOI', '10.1016/j.ccr.2004.09.011')]\n",
      "title  TrkA alternative splicing: a regulated tumor-promoting switch in human neuroblastoma.\n",
      "authors  Ronnstrand L.\n",
      "comments  []\n",
      "location  Cell. Mol. Life Sci. 61:2535-2548(2004).\n",
      "number  31\n",
      "positions  ['REVIEW ON ROLE IN KIT SIGNALING, AND PHOSPHORYLATION.']\n",
      "references  [('PubMed', '15526160'), ('DOI', '10.1007/s00018-004-4189-6')]\n",
      "title  Signal transduction via the stem cell factor receptor/c-Kit.\n",
      "authors  Audero E., Cascone I., Maniero F., Napione L., Arese M., Lanfrancone L., Bussolino F.\n",
      "comments  []\n",
      "location  J. Biol. Chem. 279:13224-13233(2004).\n",
      "number  32\n",
      "positions  ['FUNCTION, PHOSPHORYLATION, AND INTERACTION WITH TEK.']\n",
      "references  [('PubMed', '14665640'), ('DOI', '10.1074/jbc.M307456200')]\n",
      "title  Adaptor ShcA protein binds tyrosine kinase Tie2 receptor and regulates migration and sprouting but not survival of endothelial cells.\n",
      "authors  Wang J.F., Zhang X., Groopman J.E.\n",
      "comments  []\n",
      "location  J. Biol. Chem. 279:27088-27097(2004).\n",
      "number  33\n",
      "positions  ['INTERACTION WITH FLT4.']\n",
      "references  [('PubMed', '15102829'), ('DOI', '10.1074/jbc.M314015200')]\n",
      "title  Activation of vascular endothelial growth factor receptor-3 and its downstream signaling promote cell survival under oxidative stress.\n",
      "authors  Hinsby A.M., Olsen J.V., Mann M.\n",
      "comments  []\n",
      "location  J. Biol. Chem. 279:46438-46447(2004).\n",
      "number  34\n",
      "positions  ['INTERACTION WITH IRS4.']\n",
      "references  [('PubMed', '15316024'), ('DOI', '10.1074/jbc.M404537200')]\n",
      "title  Tyrosine phosphoproteomics of fibroblast growth factor signaling: a role for insulin receptor substrate-4.\n",
      "authors  Ventura A., Maccarana M., Raker V.A., Pelicci P.-G.\n",
      "comments  []\n",
      "location  J. Biol. Chem. 279:2299-2306(2004).\n",
      "number  35\n",
      "positions  ['SUBCELLULAR LOCATION (ISOFORM P46SHC).']\n",
      "references  [('PubMed', '14573619'), ('DOI', '10.1074/jbc.M307655200')]\n",
      "title  A cryptic targeting signal induces isoform-specific localization of p46Shc to mitochondria.\n",
      "authors  Ranganathan S., Liu C.-X., Migliorini M.M., Von Arnim C.A.F., Peltan I.D., Mikhailenko I., Hyman B.T., Strickland D.K.\n",
      "comments  []\n",
      "location  J. Biol. Chem. 279:40536-40544(2004).\n",
      "number  36\n",
      "positions  ['INTERACTION WITH LRP1.']\n",
      "references  [('PubMed', '15272003'), ('DOI', '10.1074/jbc.M407592200')]\n",
      "title  Serine and threonine phosphorylation of the low density lipoprotein receptor-related protein by protein kinase Calpha regulates endocytosis and association with adaptor molecules.\n",
      "authors  Smith W.W., Norton D.D., Gorospe M., Jiang H., Nemoto S., Holbrook N.J., Finkel T., Kusiak J.W.\n",
      "comments  []\n",
      "location  J. Cell Biol. 169:331-339(2005).\n",
      "number  37\n",
      "positions  ['PHOSPHORYLATION AT SER-36.']\n",
      "references  [('PubMed', '15837797'), ('DOI', '10.1083/jcb.200410041')]\n",
      "title  Phosphorylation of p66Shc and forkhead proteins mediates Abeta toxicity.\n",
      "authors  Rush J., Moritz A., Lee K.A., Guo A., Goss V.L., Spek E.J., Zhang H., Zha X.-M., Polakiewicz R.D., Comb M.J.\n",
      "comments  []\n",
      "location  Nat. Biotechnol. 23:94-101(2005).\n",
      "number  38\n",
      "positions  ['IDENTIFICATION BY MASS SPECTROMETRY [LARGE SCALE ANALYSIS].']\n",
      "references  [('PubMed', '15592455'), ('DOI', '10.1038/nbt1046')]\n",
      "title  Immunoaffinity profiling of tyrosine phosphorylation in cancer cells.\n",
      "authors  Olsen J.V., Blagoev B., Gnad F., Macek B., Kumar C., Mortensen P., Mann M.\n",
      "comments  [('TISSUE', 'Cervix carcinoma')]\n",
      "location  Cell 127:635-648(2006).\n",
      "number  39\n",
      "positions  ['PHOSPHORYLATION [LARGE SCALE ANALYSIS] AT SER-139, AND IDENTIFICATION', 'BY MASS SPECTROMETRY [LARGE SCALE ANALYSIS].']\n",
      "references  [('PubMed', '17081983'), ('DOI', '10.1016/j.cell.2006.09.026')]\n",
      "title  Global, in vivo, and site-specific phosphorylation dynamics in signaling networks.\n",
      "authors  Degoutin J., Vigny M., Gouzi J.Y.\n",
      "comments  []\n",
      "location  FEBS Lett. 581:727-734(2007).\n",
      "number  40\n",
      "positions  ['INTERACTION WITH ALK, AND PHOSPHORYLATION.']\n",
      "references  [('PubMed', '17274988'), ('DOI', '10.1016/j.febslet.2007.01.039')]\n",
      "title  ALK activation induces Shc and FRS2 recruitment: Signaling and phenotypic outcomes in PC12 cells differentiation.\n",
      "authors  Brummer T., Larance M., Herrera Abreu M.T., Lyons R.J., Timpson P., Emmerich C.H., Fleuren E.D.G., Lehrbach G.M., Schramek D., Guilhaus M., James D.E., Daly R.J.\n",
      "comments  []\n",
      "location  EMBO J. 27:2305-2316(2008).\n",
      "number  41\n",
      "positions  ['INTERACTION WITH GAB2.']\n",
      "references  [('PubMed', '19172738'), ('DOI', '10.1038/emboj.2008.159')]\n",
      "title  Phosphorylation-dependent binding of 14-3-3 terminates signalling by the Gab2 docking protein.\n",
      "authors  Dephoure N., Zhou C., Villen J., Beausoleil S.A., Bakalarski C.E., Elledge S.J., Gygi S.P.\n",
      "comments  [('TISSUE', 'Cervix carcinoma')]\n",
      "location  Proc. Natl. Acad. Sci. U.S.A. 105:10762-10767(2008).\n",
      "number  42\n",
      "positions  ['PHOSPHORYLATION [LARGE SCALE ANALYSIS] AT SER-139 AND TYR-427, AND', 'IDENTIFICATION BY MASS SPECTROMETRY [LARGE SCALE ANALYSIS].']\n",
      "references  [('PubMed', '18669648'), ('DOI', '10.1073/pnas.0805139105')]\n",
      "title  A quantitative atlas of mitotic phosphorylation.\n",
      "authors  Oppermann F.S., Gnad F., Olsen J.V., Hornberger R., Greff Z., Keri G., Mann M., Daub H.\n",
      "comments  []\n",
      "location  Mol. Cell. Proteomics 8:1751-1764(2009).\n",
      "number  43\n",
      "positions  ['IDENTIFICATION BY MASS SPECTROMETRY [LARGE SCALE ANALYSIS].']\n",
      "references  [('PubMed', '19369195'), ('DOI', '10.1074/mcp.M800588-MCP200')]\n",
      "title  Large-scale proteomics analysis of the human kinome.\n",
      "authors  Mayya V., Lundgren D.H., Hwang S.-I., Rezaul K., Wu L., Eng J.K., Rodionov V., Han D.K.\n",
      "comments  [('TISSUE', 'Leukemic T-cell')]\n",
      "location  Sci. Signal. 2:RA46-RA46(2009).\n",
      "number  44\n",
      "positions  ['PHOSPHORYLATION [LARGE SCALE ANALYSIS] AT TYR-427, AND IDENTIFICATION', 'BY MASS SPECTROMETRY [LARGE SCALE ANALYSIS].']\n",
      "references  [('PubMed', '19690332'), ('DOI', '10.1126/scisignal.2000007')]\n",
      "title  Quantitative phosphoproteomic analysis of T cell receptor signaling reveals system-wide modulation of protein-protein interactions.\n",
      "authors  Heinrich C., Keller C., Boulay A., Vecchi M., Bianchi M., Sack R., Lienhard S., Duss S., Hofsteenge J., Hynes N.E.\n",
      "comments  []\n",
      "location  Oncogene 29:1598-1610(2010).\n",
      "number  45\n",
      "positions  ['INTERACTION WITH CPNE3.']\n",
      "references  [('PubMed', '20010870'), ('DOI', '10.1038/onc.2009.456')]\n",
      "title  Copine-III interacts with ErbB2 and promotes tumor cell migration.\n",
      "authors  Olsen J.V., Vermeulen M., Santamaria A., Kumar C., Miller M.L., Jensen L.J., Gnad F., Cox J., Jensen T.S., Nigg E.A., Brunak S., Mann M.\n",
      "comments  [('TISSUE', 'Cervix carcinoma')]\n",
      "location  Sci. Signal. 3:RA3-RA3(2010).\n",
      "number  46\n",
      "positions  ['IDENTIFICATION BY MASS SPECTROMETRY [LARGE SCALE ANALYSIS].']\n",
      "references  [('PubMed', '20068231'), ('DOI', '10.1126/scisignal.2000475')]\n",
      "title  Quantitative phosphoproteomics reveals widespread full phosphorylation site occupancy during mitosis.\n",
      "authors  Burkard T.R., Planyavsky M., Kaupe I., Breitwieser F.P., Buerckstuemmer T., Bennett K.L., Superti-Furga G., Colinge J.\n",
      "comments  []\n",
      "location  BMC Syst. Biol. 5:17-17(2011).\n",
      "number  47\n",
      "positions  ['IDENTIFICATION BY MASS SPECTROMETRY [LARGE SCALE ANALYSIS].']\n",
      "references  [('PubMed', '21269460'), ('DOI', '10.1186/1752-0509-5-17')]\n",
      "title  Initial characterization of the human central proteome.\n",
      "authors  Rigbolt K.T., Prokhorova T.A., Akimov V., Henningsen J., Johansen P.T., Kratchmarova I., Kassem M., Mann M., Olsen J.V., Blagoev B.\n",
      "comments  []\n",
      "location  Sci. Signal. 4:RS3-RS3(2011).\n",
      "number  48\n",
      "positions  ['PHOSPHORYLATION [LARGE SCALE ANALYSIS] AT SER-139, AND IDENTIFICATION', 'BY MASS SPECTROMETRY [LARGE SCALE ANALYSIS].']\n",
      "references  [('PubMed', '21406692'), ('DOI', '10.1126/scisignal.2001570')]\n",
      "title  System-wide temporal characterization of the proteome and phosphoproteome of human embryonic stem cell differentiation.\n",
      "authors  Van Damme P., Lasa M., Polevoda B., Gazquez C., Elosegui-Artola A., Kim D.S., De Juan-Pardo E., Demeyer K., Hole K., Larrea E., Timmerman E., Prieto J., Arnesen T., Sherman F., Gevaert K., Aldabe R.\n",
      "comments  []\n",
      "location  Proc. Natl. Acad. Sci. U.S.A. 109:12449-12454(2012).\n",
      "number  49\n",
      "positions  ['ACETYLATION [LARGE SCALE ANALYSIS] AT MET-1, ACETYLATION [LARGE SCALE', 'ANALYSIS] AT MET-1 (ISOFORMS 7 AND P52SHC), AND IDENTIFICATION BY MASS', 'SPECTROMETRY [LARGE SCALE ANALYSIS].']\n",
      "references  [('PubMed', '22814378'), ('DOI', '10.1073/pnas.1210303109')]\n",
      "title  N-terminal acetylome analyses and functional insights of the N- terminal acetyltransferase NatB.\n",
      "authors  Strunk U., Saffran H.A., Wu F.W., Smiley J.R.\n",
      "comments  []\n",
      "location  J. Virol. 87:11276-11286(2013).\n",
      "number  50\n",
      "positions  ['INTERACTION WITH HERPES SIMPLEX VIRUS 1 UL46.']\n",
      "references  [('PubMed', '23946459'), ('DOI', '10.1128/JVI.01702-13')]\n",
      "title  Role of herpes simplex virus VP11/12 tyrosine-based motifs in binding and activation of the Src family kinase Lck and recruitment of p85, Grb2, and Shc.\n",
      "authors  Bian Y., Song C., Cheng K., Dong M., Wang F., Huang J., Sun D., Wang L., Ye M., Zou H.\n",
      "comments  [('TISSUE', 'Liver')]\n",
      "location  J. Proteomics 96:253-262(2014).\n",
      "number  51\n",
      "positions  ['PHOSPHORYLATION [LARGE SCALE ANALYSIS] AT SER-139, AND IDENTIFICATION', 'BY MASS SPECTROMETRY [LARGE SCALE ANALYSIS].']\n",
      "references  [('PubMed', '24275569'), ('DOI', '10.1016/j.jprot.2013.11.014')]\n",
      "title  An enzyme assisted RP-RPLC approach for in-depth analysis of human liver phosphoproteome.\n",
      "authors  Mikol V., Baumann G., Zurini M.G.M., Hommel U.\n",
      "comments  []\n",
      "location  J. Mol. Biol. 254:86-95(1995).\n",
      "number  52\n",
      "positions  ['X-RAY CRYSTALLOGRAPHY (2.7 ANGSTROMS) OF 482-583.']\n",
      "references  [('PubMed', '7473762'), ('DOI', '10.1006/jmbi.1995.0601')]\n",
      "title  Crystal structure of the SH2 domain from the adaptor protein SHC: a model for peptide binding based on X-ray and NMR data.\n",
      "authors  Zhou M.-M., Ravichandran K.S., Olejniczak E.F., Petros A.M., Meadows R.P., Sattler M., Harlan J.E., Wade W.S., Burakoff S.J., Fesik S.W.\n",
      "comments  []\n",
      "location  Nature 378:584-592(1995).\n",
      "number  53\n",
      "positions  ['STRUCTURE BY NMR OF 127-317.']\n",
      "references  [('PubMed', '8524391'), ('DOI', '10.1038/378584a0')]\n",
      "title  Structure and ligand recognition of the phosphotyrosine binding domain of Shc.\n",
      "authors  Zhou M.-M., Meadows R.P., Logan T.M., Yoon H.S., Wade W.S., Ravichandran K.S., Burakoff S.J., Fesik S.W.\n",
      "comments  []\n",
      "location  Proc. Natl. Acad. Sci. U.S.A. 92:7784-7788(1995).\n",
      "number  54\n",
      "positions  ['STRUCTURE BY NMR OF 480-583 IN COMPLEX WITH TYROSINE-PHOSPHORYLATED', 'CD3Z.']\n",
      "references  [('PubMed', '7544002'), ('DOI', '10.1073/pnas.92.17.7784')]\n",
      "title  Solution structure of the Shc SH2 domain complexed with a tyrosine- phosphorylated peptide from the T-cell receptor.\n"
     ]
    }
   ],
   "source": [
    "# print(dir(record.references[1]))\n",
    "for reference in record.references:\n",
    "    print(\"authors \",reference.authors)\n",
    "    print(\"comments \",reference.comments)\n",
    "    print(\"location \", reference.location)\n",
    "    print(\"number \",reference.number)    \n",
    "    print(\"positions \",reference.positions)\n",
    "    print(\"references \",reference.references)    \n",
    "    print(\"title \",reference.title)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Motifs #"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 241,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from Bio import motifs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.4.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
